<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Data conversion &mdash; MNE 0.9.dev0 documentation</title>
    
    <link rel="stylesheet" href="../_static/navy.css" type="text/css" />
    <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '../',
        VERSION:     '0.9.dev0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="../_static/jquery.js"></script>
    <script type="text/javascript" src="../_static/underscore.js"></script>
    <script type="text/javascript" src="../_static/doctools.js"></script>
    <link rel="top" title="MNE 0.9.dev0 documentation" href="../index.html" />
    <link rel="up" title="Manual" href="../manual.html" />
    <link rel="next" title="The Matlab toolbox" href="matlab.html" />
    <link rel="prev" title="Morphing and averaging" href="morph.html" />

    <script type="text/javascript">
    var _gaq = _gaq || [];
    _gaq.push(['_setAccount', 'UA-37225609-1']);
    _gaq.push(['_trackPageview']);

    (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
    })();
    </script>



    <script type="text/javascript">
    !function(d,s,id){var js,fjs=d.getElementsByTagName(s)[0];if(!d.getElementById(id)){js=d.createElement(s);
    js.id=id;js.src="http://platform.twitter.com/widgets.js";
    fjs.parentNode.insertBefore(js,fjs);}}(document,"script","twitter-wjs");
    </script>



    <script type="text/javascript">
    (function() {
    var po = document.createElement('script'); po.type = 'text/javascript'; po.async = true;
    po.src = 'https://apis.google.com/js/plusone.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(po, s);
    })();
    </script>


  </head>
  <body>

<div style="background-color: red; color: white; font-weight:bold; text-align: center; padding: 10px; min-width: 910px">
This documentation is for the development version (0.9.dev0) - <a href="http://martinos.org/mne/stable">Stable version</a>
</div>

<div style="background-color: white; text-align: left; padding: 10px 7px 15px 15px; min-width: 910px">
<div style="float: left">
<a href="../index.html"><img src="../_static/mne_logo.png" border="0" alt="py4sci"/></a>
</div>

<div style="float: right">
<a href="../index.html"><img src="../_static/institutions.png" border="0" alt="py4sci"/></a>
</div>
<br style="clear:both"/>
</div>

    <div class="related">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="../np-modindex.html" title="Python Module Index"
             >modules</a></li>
        <li class="right" >
          <a href="matlab.html" title="The Matlab toolbox"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="morph.html" title="Morphing and averaging"
             accesskey="P">previous</a> |</li>
        <li><a href="../index.html">Home</a>&nbsp;|&nbsp;</li>
        <li><a href="../manual.html">Manual</a>&nbsp;|&nbsp;</li>
        <li><a href="../mne-python.html">Python</a>&nbsp;|&nbsp;</li>
        <li><a href="../cite.html">Cite MNE</a>&nbsp;|&nbsp;</li>
        <!-- <li><a href="../search.html">Search</a></li> -->

          <li><a href="../manual.html" accesskey="U">Manual</a> &raquo;</li> 
      </ul>
    </div>

      <div class="sphinxsidebar">
        <div class="sphinxsidebarwrapper">
<h3><a href="../index.html">Table Of Contents</a></h3>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../manual.html">Manual</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="intro.html">Introduction</a></li>
<li class="toctree-l2"><a class="reference internal" href="list.html">Overview</a></li>
<li class="toctree-l2"><a class="reference internal" href="cookbook.html">The Cookbook</a></li>
<li class="toctree-l2"><a class="reference internal" href="browse.html">Processing raw data</a></li>
<li class="toctree-l2"><a class="reference internal" href="forward.html">The forward solution</a></li>
<li class="toctree-l2"><a class="reference internal" href="mne.html">The current estimates</a></li>
<li class="toctree-l2"><a class="reference internal" href="analyze.html">Interactive analysis</a></li>
<li class="toctree-l2"><a class="reference internal" href="morph.html">Morphing and averaging</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="">Data conversion</a></li>
<li class="toctree-l2"><a class="reference internal" href="matlab.html">The Matlab toolbox</a></li>
<li class="toctree-l2"><a class="reference internal" href="utilities.html">Miscellaneous utilities</a></li>
<li class="toctree-l2"><a class="reference internal" href="sampledata.html">The sample data set</a></li>
<li class="toctree-l2"><a class="reference internal" href="reading.html">Related publications</a></li>
<li class="toctree-l2"><a class="reference internal" href="AppA.html">Creating the BEM meshes</a></li>
<li class="toctree-l2"><a class="reference internal" href="AppB.html">Setup at the Martinos Center</a></li>
<li class="toctree-l2"><a class="reference internal" href="AppInstall.html">Installation and configuration</a></li>
<li class="toctree-l2"><a class="reference internal" href="AppReleaseNotes.html">Release notes</a></li>
<li class="toctree-l2"><a class="reference internal" href="AppEULA.html">Licence agreement</a></li>
<li class="toctree-l2"><a class="reference internal" href="../command_line_tutorial.html">Getting started with MNE command line</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../mne-python.html">MNE with Python</a></li>
<li class="toctree-l1"><a class="reference internal" href="../mne-cpp.html">MNE with CPP</a></li>
<li class="toctree-l1"><a class="reference internal" href="../cite.html">Cite MNE and MNE-Python</a></li>
</ul>

<div id="searchbox" style="display: none">
  <h3>Quick search</h3>
    <form class="search" action="../search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script><h3>Versions</h3>
<ul class="current">
    <li class="toctree-l1"><a href=http://martinos.org/mne/stable>Stable</a></li>
    <li class="toctree-l1"><a href=http://martinos.org/mne/dev>Development</a></li>
</ul>
        </div>
      </div>

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body">
            
  <div class="section" id="data-conversion">
<span id="ch-convert"></span><h1>Data conversion<a class="headerlink" href="#data-conversion" title="Permalink to this headline">¶</a></h1>
<div class="section" id="overview">
<h2>Overview<a class="headerlink" href="#overview" title="Permalink to this headline">¶</a></h2>
<p>This Chapter describes the data conversion utilities included
with the MNE software.</p>
</div>
<div class="section" id="importing-data-from-other-meg-eeg-systems">
<span id="behiaadg"></span><h2>Importing data from other MEG/EEG systems<a class="headerlink" href="#importing-data-from-other-meg-eeg-systems" title="Permalink to this headline">¶</a></h2>
<p>This section describes the utilities to convert data from
other MEG/EEG systems into the fif format.</p>
<div class="section" id="importing-4-d-neuroimaging-data">
<h3>Importing 4-D Neuroimaging data<a class="headerlink" href="#importing-4-d-neuroimaging-data" title="Permalink to this headline">¶</a></h3>
<p>The newest version of 4-D Magnes software includes the possibility
to export data in fif. Please consult the documentation of the Magnes
system for details of this export utility. However, the exported
fif file does not include information about the compensation channels
and the weights to be applied to realize software gradient compensation.
To augment the Magnes fif files with the necessary information,
the MNE software includes the utilities mne_insert_4D_comp , mne_create_comp_data ,
and mne_add_to_meas_info.</p>
<p>As a result, the complete 4D Magnes data conversion process
involves the following steps:</p>
<ul class="simple">
<li>Export the raw data fif file from the
Magnes system.</li>
<li>If the data comes from a Magnes system where the primary (helmet) sensors
are gradiometers instead of magnetometers, run mne_fix_mag_coil_types with
the <tt class="docutils literal"><span class="pre">--magnes</span></tt> option to correct the channel information
in the file, see <a class="reference internal" href="utilities.html#chdgaajc"><em>Updating magnetometer coil types: mne_fix_mag_coil_types</em></a>.</li>
<li>Export a text file containing the Magnes compensation sensor
data.</li>
<li>Create a text file containing the appropriate compensation
channel weights.</li>
<li>Run mne_insert_4D_comp with
the files created in the first two steps to merge compensation channel
data with the original Magnes fif file.</li>
<li>Run mne_create_comp_data on
the file created in step 3. to make a fif file containing the compensation
weights.</li>
<li>Run mne_add_to_meas_info with
the fif files created in steps 4. and 5. as input to result in a
complete fif file containing all the necessary data.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Including the compensation channel data is recommended    but not mandatory. If the data are saved in the Magnes system are    already compensated, there will be a small error in the forward    calculations whose significance has not been evaluated carefully    at this time.</p>
</div>
</div>
<div class="section" id="importing-ctf-data">
<span id="behdebch"></span><h3>Importing CTF data<a class="headerlink" href="#importing-ctf-data" title="Permalink to this headline">¶</a></h3>
<p>The MNE software includes a utility mne_ctf2fiff ,
based on the BrainStorm Matlab code by Richard Leahy, John Mosher,
and Sylvain Baillet, to convert data in CTF ds directory to fif
format.</p>
<p>The command-line options of mne_ctf2fiff are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;verbose</strong></p>
<blockquote>
<div>Produce a verbose listing of the conversion process to stdout.</div></blockquote>
<p><strong>&#8212;ds &lt;*directory*&gt;</strong></p>
<blockquote>
<div>Read the data from this directory</div></blockquote>
<p><strong>&#8212;omit &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Read the names of channels to be omitted from this text file. Enter one
channel name per line. The names should match exactly with those
listed in the CTF data structures. By default, all channels are included.</div></blockquote>
<p><strong>&#8212;fif &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of the output file. If the length of the raw data exceeds
the 2-GByte fif file limit, several output files will be produced.
These additional &#8216;extension&#8217; files will be tagged
with <tt class="docutils literal"><span class="pre">_001.fif</span></tt> , <tt class="docutils literal"><span class="pre">_002.fif</span></tt> , etc.</div></blockquote>
<p><strong>&#8212;evoked</strong></p>
<blockquote>
<div>Produce and evoked-response fif file instead of a raw data file.
Each trial in the CTF data file is included as a separate category
(condition). The maximum number of samples in each trial is limited
to 25000.</div></blockquote>
<p><strong>&#8212;infoonly</strong></p>
<blockquote>
<div>Write only the measurement info to the output file, do not include data.</div></blockquote>
<p>During conversion, the following files are consulted from
the ds directory:</p>
<p>** &lt;<em>name</em>&gt; .res4**</p>
<blockquote>
<div>This file contains most of the header information pertaining the acquisition.</div></blockquote>
<p>** &lt;<em>name</em>&gt; .hc**</p>
<blockquote>
<div>This file contains the HPI coil locations in sensor and head coordinates.</div></blockquote>
<p>** &lt;<em>name</em>&gt; .meg4**</p>
<blockquote>
<div>This file contains the actual MEG data. If the data are split across several
files due to the 2-GByte file size restriction, the &#8216;extension&#8217; files
are called &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">.</span></tt> &lt;<em>number</em>&gt; <tt class="docutils literal"><span class="pre">_meg4</span></tt> .</div></blockquote>
<p>** &lt;<em>name</em>&gt; .eeg**</p>
<blockquote>
<div>This is an optional input file containing the EEG electrode locations. More
details are given below.</div></blockquote>
<p>If the &lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">.eeg</span></tt> file,
produced from the Polhemus data file with CTF software, is present,
it is assumed to contain lines with the format:</p>
<blockquote>
<div>&lt;<em>number</em>&gt; &lt;<em>name</em>&gt; &lt;<em>x/cm</em>&gt; &lt;<em>y/cm</em>&gt; &lt;<em>z/cm</em>&gt;</div></blockquote>
<p>The field &lt;<em>number</em>&gt; is
a sequential number to be assigned to the converted data point in
the fif file. &lt;<em>name</em>&gt; is either
a name of an EEG channel, one of <tt class="docutils literal"><span class="pre">left</span></tt> , <tt class="docutils literal"><span class="pre">right</span></tt> ,
or <tt class="docutils literal"><span class="pre">nasion</span></tt> to indicate a fiducial landmark, or any word
which is not a name of any channel in the data. If &lt;<em>name</em>&gt; is
a name of an EEG channel available in the data, the location is
included in the Polhemus data as an EEG electrode locations and
inserted as the location of the EEG electrode. If the name is one
of the fiducial landmark names, the point is included in the Polhemus
data as a fiducial landmark. Otherwise, the point is included as
an additional head surface points.</p>
<p>The standard <tt class="docutils literal"><span class="pre">eeg</span></tt> file produced by CTF software
does not contain the fiducial locations. If desired, they can be
manually copied from the <tt class="docutils literal"><span class="pre">pos</span></tt> file which was the source
of the <tt class="docutils literal"><span class="pre">eeg</span></tt> file.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">In newer CTF data the EEG position information    maybe present in the <tt class="docutils literal"><span class="pre">res4</span></tt> file. If the <tt class="docutils literal"><span class="pre">eeg</span></tt> file    is present, the positions given there take precedence over the information    in the <tt class="docutils literal"><span class="pre">res4</span></tt> file.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">mne_ctf2fiff converts    both epoch mode and continuous raw data file into raw data fif files.    It is not advisable to use epoch mode files with time gaps between    the epochs because the data will be discontinuous in the resulting    fif file with jumps at the junctions between epochs. These discontinuities    produce artefacts if the raw data is filtered in mne_browse_raw , mne_process_raw ,    or graph .</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The conversion process includes a transformation    from the CTF head coordinate system convention to that used in the    Neuromag systems.</p>
</div>
</div>
<div class="section" id="importing-ctf-polhemus-data">
<span id="behbabfa"></span><h3>Importing CTF Polhemus data<a class="headerlink" href="#importing-ctf-polhemus-data" title="Permalink to this headline">¶</a></h3>
<p>The CTF MEG systems store the Polhemus digitization data
in text files. The utility mne_ctf_dig2fiff was
created to convert these data files into the fif and hpts formats.</p>
<p>The input data to mne_ctf_dig2fiff is
a text file, which contains the coordinates of the digitization
points in centimeters. The first line should contain a single number
which is the number of points listed in the file. Each of the following
lines contains a sequential number of the point, followed by the
three coordinates. mne_ctf_dig2fiff ignores
any text following the <img class="math" src="../_images/math/84d7271dd9e78c1e05d6c3c6ecb60309ef7dfc73.png" alt="z"/> coordinate
on each line. If the <tt class="docutils literal"><span class="pre">--numfids</span></tt> option is specified,
the first three points indicate the three fiducial locations (1
= nasion, 2 = left auricular point, 3 = right auricular point).
Otherwise, the input file must end with three lines beginning with <tt class="docutils literal"><span class="pre">left</span></tt> , <tt class="docutils literal"><span class="pre">right</span></tt> ,
or <tt class="docutils literal"><span class="pre">nasion</span></tt> to indicate the locations of the fiducial
landmarks, respectively.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The sequential numbers should be unique within    a file. I particular, the numbers 1, 2, and 3 must not be appear    more than once if the <tt class="docutils literal"><span class="pre">--numfids</span></tt> options is used.</p>
</div>
<p>The command-line options for mne_ctf_dig2fiff are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;dig &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the input data file in CTF output format.</div></blockquote>
<p><strong>&#8212;numfids</strong></p>
<blockquote>
<div>Fiducial locations are numbered instead of labeled, see above.</div></blockquote>
<p><strong>&#8212;hpts &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the output hpts file. The format of this text file is
described in <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a>.</div></blockquote>
<p><strong>&#8212;fif &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the output fif file.</div></blockquote>
</div>
<div class="section" id="applying-software-gradient-compensation">
<span id="behddfbi"></span><h3>Applying software gradient compensation<a class="headerlink" href="#applying-software-gradient-compensation" title="Permalink to this headline">¶</a></h3>
<p>Since the software gradient compensation employed in CTF
systems is a reversible operation, it is possible to change the
compensation status of CTF data in the data files as desired. This
section contains information about the technical details of the
compensation procedure and a description of mne_compensate_data ,
which is a utility to change the software gradient compensation
state in evoked-response data files.</p>
<p>The fif files containing CTF data converted using the utility mne_ctf2fiff contain
several compensation matrices which are employed to suppress external disturbances
with help of the reference channel data. The reference sensors are
located further away from the brain than the helmet sensors and
are thus measuring mainly the external disturbances rather than magnetic
fields originating in the brain. Most often, a compensation matrix
corresponding to a scheme nicknamed <em>Third-order gradient
compensation</em> is employed.</p>
<p>Let us assume that the data contain <img class="math" src="../_images/math/26038e5468fa3f743e11b649cd49e30d4bf2d9aa.png" alt="n_1"/> MEG
sensor channels, <img class="math" src="../_images/math/a8ab03f1f7e4f60328e1b8d96389e2d5a31e6e24.png" alt="n_2"/> reference sensor
channels, and <img class="math" src="../_images/math/e18c8ad4269906940247b5473061c1fcc99d3e3e.png" alt="n_3"/> other channels.
The data from all channels can be concatenated into a single vector</p>
<div class="math">
<p><img src="../_images/math/f79d517b8ab6a117e91868b6525d17f1159295cf.png" alt="x = [x_1^T x_2^T x_3^T]^T\ ,"/></p>
</div><p>where <img class="math" src="../_images/math/b31b1ea8e4fe3f618dedfb0e2d8c69a3df7a391d.png" alt="x_1"/>, <img class="math" src="../_images/math/272c62c44455c90173267b1bafdf7ecebd6bc316.png" alt="x_2"/>,
and <img class="math" src="../_images/math/5ea15a7136b3743ca4eebc744fa85be7b81a0c4d.png" alt="x_3"/> are the data vectors corresponding
to the MEG sensor channels, reference sensor channels, and other
channels, respectively. The data before and after compensation,
denoted here by <img class="math" src="../_images/math/17d7a25b73ef1d0a97c99ff0f7be199ed1f16490.png" alt="x_{(0)}"/> and <img class="math" src="../_images/math/019fc9503bded518a5ff889f0cf1b59f8c90a447.png" alt="x_{(k)}"/>, respectively,
are related by</p>
<div class="math">
<p><img src="../_images/math/e4d3753fea5a68dcb00f13668cc95c58f35869b9.png" alt="x_{(k)} = M_{(k)} x_{(0)}\ ,"/></p>
</div><p>where the composite compensation matrix is</p>
<div class="math">
<p><img src="../_images/math/bb57c03f2b375102d0c017e717e7d77c36b86711.png" alt="M_{(k)} = \begin{bmatrix}
I_{n_1} &amp; C_{(k)} &amp; 0 \\
0 &amp; I_{n_2} &amp; 0 \\
0 &amp; 0 &amp; I_{n_3}
\end{bmatrix}\ ."/></p>
</div><p>In the above, <img class="math" src="../_images/math/e4ab8f7cdcefa89d177848807b73bec6db231279.png" alt="C_{(k)}"/> is a <img class="math" src="../_images/math/26038e5468fa3f743e11b649cd49e30d4bf2d9aa.png" alt="n_1"/> by <img class="math" src="../_images/math/a8ab03f1f7e4f60328e1b8d96389e2d5a31e6e24.png" alt="n_2"/> compensation
data matrix corresponding to compensation &#8220;grade&#8221; <img class="math" src="../_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/>.
It is easy to see that</p>
<div class="math">
<p><img src="../_images/math/d4aced34fee267831e6e1f1f57983454ddd6e316.png" alt="M_{(k)}^{-1} = \begin{bmatrix}
I_{n_1} &amp; -C_{(k)} &amp; 0 \\
0 &amp; I_{n_2} &amp; 0 \\
0 &amp; 0 &amp; I_{n_3}
\end{bmatrix}\ ."/></p>
</div><p>To convert from compensation grade <img class="math" src="../_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> to <img class="math" src="../_images/math/3eca8557203e86160952e1c0f735f7417f3285b1.png" alt="p"/> one
can simply multiply the inverse of one compensate compensation matrix
by another and apply the product to the data:</p>
<div class="math">
<p><img src="../_images/math/f1dcdc57892de41ada92defdda22465e4569ae7e.png" alt="x_{(k)} = M_{(k)} M_{(p)}^{-1} x_{(p)}\ ."/></p>
</div><p>This operation is performed by mne_compensate_data ,
which has the following command-line options:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;in &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the input data file.</div></blockquote>
<p><strong>&#8212;out &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the output data file.</div></blockquote>
<p><strong>&#8212;grad &lt;*number*&gt;</strong></p>
<blockquote>
<div>Specifies the desired compensation grade in the output file. The value
can be 1, 2, 3, or 101. The values starting from 101 will be used
for 4D Magnes compensation matrices.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Only average data is included in the output.    Evoked-response data files produced with mne_browse_raw or mne_process_raw may    include standard errors of mean, which can not be re-compensated    using the above method and are thus omitted.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Raw data cannot be compensated using mne_compensate_data .    For this purpose, load the data to mne_browse_raw or mne_process_raw , specify    the desired compensation grade, and save a new raw data file.</p>
</div>
</div>
<div class="section" id="importing-magnes-compensation-channel-data">
<span id="behgddbh"></span><h3>Importing Magnes compensation channel data<a class="headerlink" href="#importing-magnes-compensation-channel-data" title="Permalink to this headline">¶</a></h3>
<p>At present, it is not possible to include reference channel
data to fif files containing 4D Magnes data directly using the conversion
utilities available for the Magnes systems. However, it is possible
to export the compensation channel signals in text format and merge
them with the MEG helmet channel data using mne_insert_4D_comp .
This utility has the following command-line options:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;in &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the input fif file containing the helmet sensor data.</div></blockquote>
<p><strong>&#8212;out &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the output fif file which will contain both the helmet
sensor data and the compensation channel data.</div></blockquote>
<p><strong>&#8212;ref &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a text file containing the reference sensor data.</div></blockquote>
<p>Each line of the reference sensor data file contains the
following information:</p>
<p><strong>epoch #</strong></p>
<blockquote>
<div>is
always one,</div></blockquote>
<p><strong>time/s</strong></p>
<blockquote>
<div>time point of this sample,</div></blockquote>
<p><strong>data/T</strong></p>
<blockquote>
<div>the reference channel data
values.</div></blockquote>
<p>The standard locations of the MEG (helmet) and compensation
sensors in a Magnes WH3600 system are listed in <tt class="docutils literal"><span class="pre">$MNE_ROOT/share/mne/Magnes_WH3600.pos</span></tt> . mne_insert_4D_comp matches
the helmet sensor positions in this file with those present in the
input data file and transforms the standard compensation channel
locations accordingly to be included in the output. Since a standard
position file is only provided for Magnes WH600, mne_insert_4D_comp only
works for that type of a system.</p>
<p>The fif files exported from the Magnes systems may contain
slightly smaller number of samples than originally acquired because
the total number of samples may not be evenly divisible with a reasonable
number of samples which will be used as the fif raw data file buffer
size. Therefore, the reference channel data may contain more samples
than the fif file. The superfluous samples will be omitted from
the end.</p>
</div>
<div class="section" id="creating-software-gradient-compensation-data">
<span id="behbiiff"></span><h3>Creating software gradient compensation data<a class="headerlink" href="#creating-software-gradient-compensation-data" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_create_comp_data was
written to create software gradient compensation weight data for
4D Magnes fif files. This utility takes a text file containing the
compensation data as input and writes the corresponding fif file
as output. This file can be merged into the fif file containing
4D Magnes data with the utility mne_add_to_meas_info .</p>
<p>The command line options of mne_create_comp_data are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;in &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the input text file containing the compensation data.</div></blockquote>
<p><strong>&#8212;kind &lt;*value*&gt;</strong></p>
<blockquote>
<div>The compensation type to be stored in the output file with the data. This
value defaults to 101 for the Magnes compensation and does not need
to be changed.</div></blockquote>
<p><strong>&#8212;out &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the output fif file containing the compensation channel weight
matrix <img class="math" src="../_images/math/e4ab8f7cdcefa89d177848807b73bec6db231279.png" alt="C_{(k)}"/>, see <a class="reference internal" href="#behddfbi"><em>Applying software gradient compensation</em></a>.</div></blockquote>
<p>The format of the text-format compensation data file is:</p>
<blockquote>
<div>&lt;<em>number of MEG helmet channels</em>&gt; &lt;<em>number of compensation channels included</em>&gt;
&lt;<em>cname_1</em>&gt; &lt;<em>cname_2</em>&gt; ...
&lt;<em>name_1</em>&gt; &lt;<em>weights</em>&gt;
&lt;<em>name_2</em>&gt; &lt;<em>weights</em>&gt; ...</div></blockquote>
<p>In the above &lt;<em>name_k</em>&gt; denote
names of MEG helmet channels and &lt;<em>cname_k</em>&gt;
those of the compensation channels, respectively. If the channel
names contain spaces, they must be surrounded by quotes, for example, <tt class="docutils literal"><span class="pre">&quot;MEG</span> <span class="pre">0111&quot;</span></tt> .</p>
</div>
<div class="section" id="importing-kit-meg-system-data">
<span id="behbjggf"></span><h3>Importing KIT MEG system data<a class="headerlink" href="#importing-kit-meg-system-data" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_kit2fiff was
created in collaboration with Alec Maranz and Asaf Bachrach to import
their MEG data acquired with the 160-channel KIT MEG system to MNE
software.</p>
<p>To import the data, the following input files are mandatory:</p>
<ul class="simple">
<li>The Polhemus data file (elp file)
containing the locations of the fiducials and the head-position
indicator (HPI) coils. These data are usually given in the CTF/4D
head coordinate system. However, mne_kit2fiff does
not rely on this assumption. This file can be exported directly from
the KIT system.</li>
<li>A file containing the locations of the HPI coils in the MEG
device coordinate system. These data are used together with the elp file
to establish the coordinate transformation between the head and
device coordinate systems. This file can be produced easily by manually
editing one of the files exported by the KIT system.</li>
<li>A sensor data file (sns file)
containing the locations and orientations of the sensors. This file
can be exported directly from the KIT system.</li>
</ul>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The output fif file will use the Neuromag head    coordinate system convention, see <a class="reference internal" href="forward.html#bjebibai"><em>The head and device coordinate systems</em></a>. A coordinate    transformation between the CTF/4D head coordinates and the Neuromag    head coordinates is included. This transformation can be read with    MNE Matlab Toolbox routines, see <a class="reference internal" href="matlab.html#ch-matlab"><em>The Matlab toolbox</em></a>.</p>
</div>
<p>The following input files are optional:</p>
<ul class="simple">
<li>A head shape data file (hsp file)
containing locations of additional points from the head surface.
These points must be given in the same coordinate system as that
used for the elp file and the
fiducial locations must be within 1 mm from those in the elp file.</li>
<li>A raw data file containing the raw data values, sample by
sample, as text. If this file is not specified, the output fif file
will only contain the measurement info block.</li>
</ul>
<p>By default mne_kit2fiff includes
the first 157 channels, assumed to be the MEG channels, in the output
file. The compensation channel data are not converted by default
but can be added, together with other channels, with the <tt class="docutils literal"><span class="pre">--type</span></tt> .
The channels from 160 onwards are designated as miscellaneous input
channels (MISC 001, MISC 002, etc.). The channel names and types
of these channels can be afterwards changed with the mne_rename_channels utility,
see <a class="reference internal" href="utilities.html#chdcfeaj"><em>Modifying channel names and types: mne_rename_channels</em></a>. In addition, it is possible to synthesize
the digital trigger channel (STI 014) from available analog
trigger channel data, see the <tt class="docutils literal"><span class="pre">--stim</span></tt> option, below.
The synthesized trigger channel data value at sample <img class="math" src="../_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> will
be:</p>
<div class="math">
<p><img src="../_images/math/136aae7415c7155200337c02926c53df1a7446fb.png" alt="s(k) = \sum_{p = 1}^n {t_p(k) 2^{p - 1}}\ ,"/></p>
</div><p>where <img class="math" src="../_images/math/f13714188ac9f07e16394d15b349ba452778cf4d.png" alt="t_p(k)"/> are the thresholded
from the input channel data d_p(k):</p>
<div class="math">
<p><img src="../_images/math/3c6b90860e78a0325f1261981e3de290197c0a4a.png" alt="t_p(k) = \Bigg\{ \begin{array}{l}
    0 \text{  if  } d_p(k) \leq t\\
    1 \text{  if  } d_p(k) &gt; t
\end{array}\ ."/></p>
</div><p>The threshold value <img class="math" src="../_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> can
be adjusted with the <tt class="docutils literal"><span class="pre">--stimthresh</span></tt> option, see below.</p>
<p>mne_kit2fiff accepts
the following command-line options:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;elp &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of the file containing the locations of the fiducials and
the HPI coils. This option is mandatory.</div></blockquote>
<p><strong>&#8212;hsp &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of the file containing the locations of the fiducials and additional
points on the head surface. This file is optional.</div></blockquote>
<p><strong>&#8212;sns &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of file containing the sensor locations and orientations. This
option is mandatory.</div></blockquote>
<p><strong>&#8212;hpi &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of a text file containing the locations of the HPI coils
in the MEG device coordinate frame, given in millimeters. The order of
the coils in this file does not have to be the same as that in the elp file.
This option is mandatory.</div></blockquote>
<p><strong>&#8212;raw &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the raw data file. If this file is not specified, the
output fif file will only contain the measurement info block.</div></blockquote>
<p><strong>&#8212;sfreq &lt;*value/Hz*&gt;</strong></p>
<blockquote>
<div>The sampling frequency of the data. If this option is not specified, the
sampling frequency defaults to 1000 Hz.</div></blockquote>
<p><strong>&#8212;lowpass &lt;*value/Hz*&gt;</strong></p>
<blockquote>
<div>The lowpass filter corner frequency used in the data acquisition.
If not specified, this value defaults to 200 Hz.</div></blockquote>
<p><strong>&#8212;highpass &lt;*value/Hz*&gt;</strong></p>
<blockquote>
<div>The highpass filter corner frequency used in the data acquisition.
If not specified, this value defaults to 0 Hz (DC recording).</div></blockquote>
<p><strong>&#8212;out &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the output fif format data file. If this file
is not specified, no output is produced but the elp , hpi ,
and hsp files are processed normally.</div></blockquote>
<p><strong>&#8212;stim &lt;*chs*&gt;</strong></p>
<blockquote>
<div>Specifies a colon-separated list of numbers of channels to be used
to synthesize a digital trigger channel. These numbers refer to
the scanning order channels as listed in the sns file,
starting from one. The digital trigger channel will be the last
channel in the file. If this option is absent, the output file will
not contain a trigger channel.</div></blockquote>
<p><strong>&#8212;stimthresh &lt;*value*&gt;</strong></p>
<blockquote>
<div>The threshold value used when synthesizing the digital trigger channel,
see above. Defaults to 1.0.</div></blockquote>
<p><strong>&#8212;add &lt;*chs*&gt;</strong></p>
<blockquote>
<div>Specifies a colon-separated list of numbers of channels to include between
the 157 default MEG channels and the digital trigger channel. These
numbers refer to the scanning order channels as listed in the sns file,
starting from one.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The mne_kit2fiff utility    has not been extensively tested yet.</p>
</div>
</div>
<div class="section" id="importing-eeg-data-saved-in-the-edf-edf-or-bdf-format">
<span id="babhdbbd"></span><h3>Importing EEG data saved in the EDF, EDF+, or BDF format<a class="headerlink" href="#importing-eeg-data-saved-in-the-edf-edf-or-bdf-format" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id1">
<h4>Overview<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h4>
<p>The mne_edf2fiff allows
conversion of EEG data from EDF, EDF+, and BDF formats to the fif
format. Documentation for these three input formats can be found
at:</p>
<p><strong>EDF:</strong></p>
<blockquote>
<div><a class="reference external" href="http://www.edfplus.info/specs/edf.html">http://www.edfplus.info/specs/edf.html</a></div></blockquote>
<p><strong>EDF+:</strong></p>
<blockquote>
<div><a class="reference external" href="http://www.edfplus.info/specs/edfplus.html">http://www.edfplus.info/specs/edfplus.html</a></div></blockquote>
<p><strong>BDF:</strong></p>
<blockquote>
<div><a class="reference external" href="http://www.biosemi.com/faq/file_format.htm">http://www.biosemi.com/faq/file_format.htm</a></div></blockquote>
<p>EDF (European Data Format) and EDF+ are 16-bit formats while
BDF is a 24-bit variant of this format used by the EEG systems manufactured
by a company called BioSemi.</p>
<p>None of these formats support electrode location information
and  head shape digitization information. Therefore, this information
has to be provided separately. Presently hpts and elp file formats
are supported to include digitization data. For information on these
formats, see <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a> and <a class="reference external" href="http://www.sourcesignal.com/formats_probe.html">http://www.sourcesignal.com/formats_probe.html</a>.
Note that it is mandatory to have the three fiducial locations (nasion
and the two auricular points) included in the digitization data.
Using the locations of the fiducial points the digitization data
are converted to the MEG head coordinate system employed in the
MNE software, see <a class="reference internal" href="forward.html#bjebibai"><em>The head and device coordinate systems</em></a>. In the comparison of the
channel names only the initial segment up to the first &#8216;-&#8216; (dash)
in the EDF/EDF+/BDF channel name is significant.</p>
<p>The EDF+ files may contain an annotation channel which can
be used to store trigger information. The Time-stamped Annotation
Lists (TALs) on the annotation  data can be converted to a trigger
channel (STI 014) using an annotation map file which associates
an annotation label with a number on the trigger channel. The TALs
can be listed with the <tt class="docutils literal"><span class="pre">--listtal</span></tt> option,
see below.</p>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">The data samples in a BDF file    are represented in a 3-byte (24-bit) format. Since 3-byte raw data    buffers are not presently supported in the fif format    these data will be changed to 4-byte integers in the conversion.    Since the maximum size of a fif file is 2 GBytes, the maximum size of    a BDF file to be converted is approximately 1.5 GBytes</p>
</div>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">The EDF/EDF+/BDF formats support channel    dependent sampling rates. This feature is not supported by mne_edf2fiff .    However, the annotation channel in the EDF+ format can have a different    sampling rate. The annotation channel data is not included in the    fif files output.</p>
</div>
</div>
<div class="section" id="using-mne-edf2fiff">
<h4>Using mne_edf2fiff<a class="headerlink" href="#using-mne-edf2fiff" title="Permalink to this headline">¶</a></h4>
<p>The command-line options of mne_edf2fiff are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;edf &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the raw data file to process.</div></blockquote>
<p><strong>&#8212;tal &lt;*filename*&gt;</strong></p>
<blockquote>
<div>List the time-stamped annotation list (TAL) data from an EDF+ file here.
This output is useful to assist in creating the annotation map file,
see the <tt class="docutils literal"><span class="pre">--annotmap</span></tt> option, below.
This output file is an event file compatible with mne_browse_raw and mne_process_raw ,
see <a class="reference internal" href="browse.html#ch-browse"><em>Processing raw data</em></a>. In addition, in the mapping between TAL
labels and trigger numbers provided by the <tt class="docutils literal"><span class="pre">--annotmap</span></tt> option is
employed to assign trigger numbers in the event file produced. In
the absence of the <tt class="docutils literal"><span class="pre">--annotmap</span></tt> option default trigger number 1024
is used.</div></blockquote>
<p><strong>&#8212;annotmap &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specify a file which maps the labels of the TALs to numbers on a trigger
channel (STI 014) which will be added to the output file if this
option is present. This annotation map file
may contain comment lines starting with the &#8216;%&#8217; or &#8216;#&#8217; characters.
The data lines contain a label-number pair, separated by a colon.
For example, a line &#8216;Trigger-1:9&#8217; means that each
annotation labeled with the text &#8216;Trigger-1&#8217; will
be translated to the number 9 on the trigger channel.</div></blockquote>
<p><strong>&#8212;elp &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the an electrode location file. This file
is in the &#8220;probe&#8221; file format used by the <em>Source
Signal Imaging, Inc.</em> software. For description of the
format, see <a class="reference external" href="http://www.sourcesignal.com/formats_probe.html">http://www.sourcesignal.com/formats_probe.html</a>. Note
that some other software packages may produce electrode-position
files with the elp ending not
conforming to the above specification. As discussed above, the fiducial
marker locations, optional in the &#8220;probe&#8221; file
format specification are mandatory for mne_edf2fiff .
When this option is encountered on the command line any previously
specified hpts file will be ignored.</div></blockquote>
<p><strong>&#8212;hpts &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of an electrode position file in  the hpts format discussed
in <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a>. The mandatory entries are the fiducial marker
locations and the EEG electrode locations. It is recommended that
electrode (channel) names instead of numbers are used to label the
EEG electrode locations. When this option is encountered on the
command line any previously specified elp file
will be ignored.</div></blockquote>
<p><strong>&#8212;meters</strong></p>
<blockquote>
<div>Assumes that the digitization data in an hpts file
is given in meters instead of millimeters.</div></blockquote>
<p><strong>&#8212;fif &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the fif file to be output.</div></blockquote>
</div>
<div class="section" id="post-conversion-tasks">
<h4>Post-conversion tasks<a class="headerlink" href="#post-conversion-tasks" title="Permalink to this headline">¶</a></h4>
<p>This section outlines additional steps to be taken to use
the EDF/EDF+/BDF file is converted to the fif format in MNE:</p>
<ul class="simple">
<li>Some of the channels may not have a
digitized electrode location associated with them. If these channels
are used for EOG or EMG measurements, their channel types should
be changed to the correct ones using the mne_rename_channels utility,
see <a class="reference internal" href="utilities.html#chdcfeaj"><em>Modifying channel names and types: mne_rename_channels</em></a>. EEG channels which do not have a location
associated with them should be assigned to be MISC channels.</li>
<li>After the channel types are correctly defined, a topographical
layout file can be created for mne_browse_raw and mne_analyze using
the mne_make_eeg_layout utility,
see <a class="reference internal" href="utilities.html#chddgdja"><em>Creating a custom EEG layout</em></a>.</li>
<li>The trigger channel name in BDF files is &#8220;Status&#8221;.
This must be specified with the <tt class="docutils literal"><span class="pre">--digtrig</span></tt> option or with help of
the MNE_TRIGGER_CH_NAME environment variable when mne_browse_raw or mne_process_raw is
invoked, see <a class="reference internal" href="browse.html#babbgjea"><em>Common options</em></a>.</li>
<li>Only the two least significant bytes on the &#8220;Status&#8221; channel
of BDF files are significant as trigger information the <tt class="docutils literal"><span class="pre">--digtrigmask</span></tt>
0xff option MNE_TRIGGER_CH_MASK environment variable should be used
to specify this to mne_browse_raw and mne_process_raw ,
see <a class="reference internal" href="browse.html#babbgjea"><em>Common options</em></a>.</li>
</ul>
</div>
</div>
<div class="section" id="importing-eeg-data-saved-in-the-tufts-university-format">
<span id="behdgaij"></span><h3>Importing EEG data saved in the Tufts University format<a class="headerlink" href="#importing-eeg-data-saved-in-the-tufts-university-format" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_tufts2fiff was
created in collaboration with Phillip Holcomb and Annette Schmid
from Tufts University to import their EEG data to the MNE software.</p>
<p>The Tufts EEG data is included in three files:</p>
<ul class="simple">
<li>The raw data file containing the acquired
EEG data. The name of this file ends with the suffix <tt class="docutils literal"><span class="pre">.raw</span></tt> .</li>
<li>The calibration raw data file. This file contains known calibration
signals and is required to bring the data to physical units. The
name of this file ends with the suffix <tt class="docutils literal"><span class="pre">c.raw</span></tt> .</li>
<li>The electrode location information file. The name of this
file ends with the suffix <tt class="docutils literal"><span class="pre">.elp</span></tt> .</li>
</ul>
<p>The utility mne_tufts2fiff has
the following command-line options:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;raw &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the raw data file to process.</div></blockquote>
<p><strong>&#8212;cal &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of the calibration data file. If calibration data are missing, the
calibration coefficients will be set to unity.</div></blockquote>
<p><strong>&#8212;elp &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of the electrode location file. If this file is missing,
the electrode locations will be unspecified. This file is in the &#8220;probe&#8221; file
format used by the <em>Source Signal Imaging, Inc.</em> software.
For description of the format, see <a class="reference external" href="http://www.sourcesignal.com/formats_probe.html">http://www.sourcesignal.com/formats_probe.html</a>.
The fiducial marker locations, optional in the &#8220;probe&#8221; file
format specification are mandatory for mne_tufts2fiff . Note
that some other software packages may produce electrode-position
files with the elp ending not
conforming to the above specification.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The conversion process includes a transformation from the Tufts head coordinate system convention to that used in    the Neuromag systems.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The fiducial landmark locations, optional in the probe file format, must be present for mne_tufts2fiff .</p>
</div>
</div>
<div class="section" id="importing-brainvision-eeg-data">
<span id="behcccdc"></span><h3>Importing BrainVision EEG data<a class="headerlink" href="#importing-brainvision-eeg-data" title="Permalink to this headline">¶</a></h3>
<p>The utility mne_brain_vision2fiff was
created to import BrainVision EEG data. This utility also helps
to import the eXimia (Nexstim) TMS-compatible EEG system data to
the MNE software. The utility uses an optional fif file containing
the head digitization data to allow source modeling. The MNE Matlab
toolbox contains the function fiff_write_dig_file to
write a digitization file based on digitization data available in
another format, see <a class="reference internal" href="matlab.html#ch-matlab"><em>The Matlab toolbox</em></a>.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">mne_brain_vision2fiff reads events from the <tt class="docutils literal"><span class="pre">vmrk</span></tt> file referenced in the
<tt class="docutils literal"><span class="pre">vhdr</span></tt> file, but it only includes events whose &#8220;Type&#8221; is <tt class="docutils literal"><span class="pre">Stimulus</span></tt> and
whose &#8220;description&#8221; is given by <tt class="docutils literal"><span class="pre">S&lt;number&gt;</span></tt>. All other events are ignored.</p>
</div>
<p>The command-line options of mne_brain_vision2fiff are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;header &lt;*name*&gt;</strong></p>
<blockquote>
<div>The name of the BrainVision header file. The extension of this file
is <tt class="docutils literal"><span class="pre">vhdr</span></tt> . The header file typically refers to a marker
file (<tt class="docutils literal"><span class="pre">vmrk</span></tt> ) which is automatically processed and a
digital trigger channel (STI 014) is formed from the marker information.
The <tt class="docutils literal"><span class="pre">vmrk</span></tt> file is ignored if the <tt class="docutils literal"><span class="pre">--eximia</span></tt> option
is present.</div></blockquote>
<p><strong>&#8212;dig &lt;*name*&gt;</strong></p>
<blockquote>
<div>The name of the fif file containing the digitization data.</div></blockquote>
<p><strong>&#8212;orignames</strong></p>
<blockquote>
<div>Use the original EEG channel labels. If this option is absent the EEG
channels will be automatically renamed to EEG 001, EEG 002, <em>etc.</em></div></blockquote>
<p><strong>&#8212;eximia</strong></p>
<blockquote>
<div>Interpret this as an eXimia data file. The first three channels
will be thresholded and interpreted as trigger channels. The composite
digital trigger channel will be composed in the same way as in the mne_kit2fiff utility,
see <a class="reference internal" href="#behbjggf"><em>Importing KIT MEG system data</em></a>, above. In addition, the fourth channel
will be assigned as an EOG channel. This option is normally used
by the mne_eximia2fiff script,
see <a class="reference internal" href="#behgcehh"><em>Converting eXimia EEG data</em></a>.</div></blockquote>
<p><strong>&#8212;split &lt;*size/MB*&gt;</strong></p>
<blockquote>
<div>Split the output data into several files which are no more than &lt;<em>size</em>&gt; MB.
By default, the output is split into files which are just below
2 GB so that the fif file maximum size is not exceeded.</div></blockquote>
<p><strong>&#8212;out &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the output fif format data file. If &lt;<em>filename</em>&gt; ends
with <tt class="docutils literal"><span class="pre">.fif</span></tt> or <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> , these endings are
deleted. After these modifications, <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> is inserted
after the remaining part of the file name. If the file is split
into multiple parts, the additional parts will be called
&lt;<em>name</em>&gt; <tt class="docutils literal"><span class="pre">-</span></tt> &lt;<em>number</em>&gt; <tt class="docutils literal"><span class="pre">_raw.fif</span></tt> .</div></blockquote>
</div>
<div class="section" id="converting-eximia-eeg-data">
<span id="behgcehh"></span><h3>Converting eXimia EEG data<a class="headerlink" href="#converting-eximia-eeg-data" title="Permalink to this headline">¶</a></h3>
<p>EEG data from the Nexstim eXimia system can be converted
to the fif format with help of the mne_eximia2fiff script.
It creates a BrainVision <tt class="docutils literal"><span class="pre">vhdr</span></tt> file and calls mne_brain_vision2fiff.
Usage:</p>
<p><tt class="docutils literal"><span class="pre">mne_eximia2fiff</span></tt> [<tt class="docutils literal"><span class="pre">--dig</span></tt> dfile ] [<tt class="docutils literal"><span class="pre">--orignames</span></tt> ] file1 file2 ...</p>
<p>where file1 file2 ...
are eXimia <tt class="docutils literal"><span class="pre">nxe</span></tt> files and the <tt class="docutils literal"><span class="pre">--orignames</span></tt> option
is passed on to mne_brain_vision2fiff .
If you want to convert all data files in a directory, say</p>
<p><tt class="docutils literal"><span class="pre">mne_eximia2fiff</span> <span class="pre">*.nxe</span></tt></p>
<p>The optional file specified with the <tt class="docutils literal"><span class="pre">--dig</span></tt> option is assumed
to contain digitizer data from the recording in the Nexstim format.
The resulting fif data file will contain these data converted to
the fif format as well as the coordinate transformation between
the eXimia digitizer and MNE head coordinate systems.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This script converts raw data files only.</p>
</div>
</div>
</div>
<div class="section" id="converting-digitization-data">
<span id="babcjead"></span><h2>Converting digitization data<a class="headerlink" href="#converting-digitization-data" title="Permalink to this headline">¶</a></h2>
<p>The mne_convert_dig_data utility
converts Polhemus digitization data between different file formats.
The input formats are:</p>
<p><strong>fif</strong></p>
<blockquote>
<div>The
standard format used in MNE. The digitization data are typically
present in the measurement files.</div></blockquote>
<p><strong>hpts</strong></p>
<blockquote>
<div>A text format which is a translation
of the fif format data, see <a class="reference internal" href="#cjadjebh"><em>The hpts format</em></a> below.</div></blockquote>
<p><strong>elp</strong></p>
<blockquote>
<div>A text format produced by the <em>Source
Signal Imaging, Inc.</em> software. For description of this &#8220;probe&#8221; format,
see <a class="reference external" href="http://www.sourcesignal.com/formats_probe.html">http://www.sourcesignal.com/formats_probe.html</a>.</div></blockquote>
<p>The data can be output in fif and hpts formats.
Only the last command-line option specifying an input file will
be honored. Zero or more output file options can be present on the
command line.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The elp and hpts input    files may contain textual EEG electrode labels. They will not be    copied to the fif format output.</p>
</div>
<p>The command-line options of mne_convert_dig_data are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;fif &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of an input fif file.</div></blockquote>
<p><strong>&#8212;hpts &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of an input hpts file.</div></blockquote>
<p><strong>&#8212;elp &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of an input elp file.</div></blockquote>
<p><strong>&#8212;fifout &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of an output fif file.</div></blockquote>
<p><strong>&#8212;hptsout &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of an output hpts file.</div></blockquote>
<p><strong>&#8212;headcoord</strong></p>
<blockquote>
<div>The fif and hpts input
files are assumed to contain data in the  MNE head coordinate system,
see <a class="reference internal" href="forward.html#bjebibai"><em>The head and device coordinate systems</em></a>. With this option present, the data are
transformed to the MNE head coordinate system with help of the fiducial
locations in the data. Use this option if this is not the case or
if you are unsure about the definition of the coordinate system
of the fif and hpts input
data. This option is implied with elp input
files. If this option is present, the fif format output file will contain
the transformation between the original digitizer data coordinates
the MNE head coordinate system.</div></blockquote>
<div class="section" id="the-hpts-format">
<span id="cjadjebh"></span><h3>The hpts format<a class="headerlink" href="#the-hpts-format" title="Permalink to this headline">¶</a></h3>
<p>The hpts format digitzer
data file may contain comment lines starting with the pound sign
(#) and data lines of the form:</p>
<blockquote>
<div>&lt;<em>category</em>&gt; &lt;<em>identifier</em>&gt; &lt;<em>x/mm</em>&gt; &lt;<em>y/mm</em>&gt; &lt;<em>z/mm</em>&gt;</div></blockquote>
<p>where</p>
<p>** &lt;<em>category</em>&gt;**</p>
<blockquote>
<div>defines the type of points. Allowed categories are: hpi , cardinal (fiducial ),eeg ,
and extra corresponding to head-position
indicator coil locations, cardinal landmarks, EEG electrode locations,
and additional head surface points, respectively. Note that tkmedit does not
recognize the fiducial as an
alias for cardinal .</div></blockquote>
<p>** &lt;<em>identifier</em>&gt;**</p>
<blockquote>
<div>identifies the point. The identifiers are usually sequential numbers. For
cardinal landmarks, 1 = left auricular point, 2 = nasion, and 3
= right auricular point. For EEG electrodes, identifier = 0 signifies
the reference electrode. Some programs (not tkmedit )
accept electrode labels as identifiers in the eeg category.</div></blockquote>
<p>** &lt;<em>x/mm</em>&gt; , &lt;<em>y/mm</em>&gt; , &lt;<em>z/mm</em>&gt;**</p>
<blockquote>
<div>Location of the point, usually in the MEG head coordinate system, see <a class="reference internal" href="forward.html#bjebibai"><em>The head and device coordinate systems</em></a>.
Some programs have options to accept coordinates in meters instead
of millimeters. With <tt class="docutils literal"><span class="pre">--meters</span></tt> option, mne_transform_points lists
the coordinates in meters.</div></blockquote>
</div>
</div>
<div class="section" id="converting-volumetric-data-into-an-mri-overlay">
<span id="behdejec"></span><h2>Converting volumetric data into an MRI overlay<a class="headerlink" href="#converting-volumetric-data-into-an-mri-overlay" title="Permalink to this headline">¶</a></h2>
<p>With help of the mne_volume_source_space utility
(<a class="reference internal" href="forward.html#bjefehji"><em>Creating a volumetric or discrete source space</em></a>) it is possible to create a source space which
is defined within a volume rather than a surface. If the <tt class="docutils literal"><span class="pre">--mri</span></tt> option
was used in mne_volume_source_space , the
source space file contains an interpolator matrix which performs
a trilinear interpolation into the voxel space of the MRI volume
specified.</p>
<p>At present, the MNE software does not include facilities
to compute volumetric source estimates. However, it is possible
to calculate forward solutions in the volumetric grid and use the
MNE Matlab toolbox to read the forward solution. It is then possible
to compute, <em>e.g.</em>, volumetric beamformer solutions
in Matlab and output the results into w or stc files.
The purpose of the mne_volume_data2mri is
to produce MRI overlay data compatible with FreeSurfer MRI viewers
(in the mgh or mgz formats) from this type of w or stc files.</p>
<p>mne_volume_data2mri accepts
the following command-line options:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;src &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of the volumetric source space file created with mne_volume_source_space .
The source space must have been created with the <tt class="docutils literal"><span class="pre">--mri</span></tt> option,
which adds the appropriate sparse trilinear interpolator matrix
to the source space.</div></blockquote>
<p><strong>&#8212;w &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of a w file to convert
into an MRI overlay.</div></blockquote>
<p><strong>&#8212;stc &lt;*filename*&gt;</strong></p>
<blockquote>
<div>The name of the stc file to convert
into an MRI overlay. If this file has many time frames, the output
file may be huge. Note: If both <tt class="docutils literal"><span class="pre">-w</span></tt> and <tt class="docutils literal"><span class="pre">--stc</span></tt> are
specified, <tt class="docutils literal"><span class="pre">-w</span></tt> takes precedence.</div></blockquote>
<p><strong>&#8212;scale &lt;*number*&gt;</strong></p>
<blockquote>
<div>Multiply the stc or w by
this scaling constant before producing the overlay.</div></blockquote>
<p><strong>&#8212;out &lt;*filename*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the output MRI overlay file. The name must end
with either <tt class="docutils literal"><span class="pre">.mgh</span></tt> or <tt class="docutils literal"><span class="pre">.mgz</span></tt> identifying the
uncompressed and compressed FreeSurfer MRI formats, respectively.</div></blockquote>
</div>
<div class="section" id="listing-source-space-data">
<span id="behbhidh"></span><h2>Listing source space data<a class="headerlink" href="#listing-source-space-data" title="Permalink to this headline">¶</a></h2>
<p>The utility mne_list_source_space outputs
the source space information into text files suitable for loading
into the Neuromag MRIlab software.</p>
<p>The command-line options are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;src &lt;*name*&gt;</strong></p>
<blockquote>
<div>The source space to be listed. This can be either the output from mne_make_source_space
(<cite>*src.fif</cite>), output from the forward calculation (<cite>*fwd.fif</cite>), or
the output from the inverse operator decomposition (<cite>*inv.fif</cite>).</div></blockquote>
<p><strong>&#8212;mri &lt;*name*&gt;</strong></p>
<blockquote>
<div>A file containing the transformation between the head and MRI coordinates
is specified with this option. This file can be either a Neuromag
MRI description file, the output from the forward calculation (<cite>*fwd.fif</cite>),
or the output from the inverse operator decomposition (<cite>*inv.fif</cite>).
If this file is included, the output will be in head coordinates.
Otherwise the source space will be listed in MRI coordinates.</div></blockquote>
<p><strong>&#8212;dip &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the &#8216;stem&#8217; for the Neuromag text format
dipole files to be output. Two files will be produced: &lt;<em>stem</em>&gt; -lh.dip
and &lt;<em>stem</em>&gt; -rh.dip. These correspond
to the left and right hemisphere part of the source space, respectively.
This source space data can be imported to MRIlab through the File/Import/Dipoles menu
item.</div></blockquote>
<p><strong>&#8212;pnt &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the &#8216;stem&#8217; for Neuromag text format
point files to be output. Two files will be produced: &lt;<em>stem</em>&gt; -lh.pnt
and &lt;<em>stem</em>&gt; -rh.pnt. These correspond
to the left and right hemisphere part of the source space, respectively.
This source space data can be imported to MRIlab through the File/Import/Strings menu
item.</div></blockquote>
<p><strong>&#8212;exclude &lt;*name*&gt;</strong></p>
<blockquote>
<div>Exclude the source space points defined by the given FreeSurfer &#8216;label&#8217; file
from the output. The name of the file should end with <tt class="docutils literal"><span class="pre">-lh.label</span></tt>
if it refers to the left hemisphere and with <tt class="docutils literal"><span class="pre">-rh.label</span></tt> if
it lists points in the right hemisphere, respectively.</div></blockquote>
<p><strong>&#8212;include &lt;*name*&gt;</strong></p>
<blockquote>
<div>Include only the source space points defined by the given FreeSurfer &#8216;label&#8217; file
to the output. The file naming convention is the same as described
above under the <tt class="docutils literal"><span class="pre">--exclude</span></tt> option. Are &#8216;include&#8217; labels are
processed before the &#8216;exclude&#8217; labels.</div></blockquote>
<p><strong>&#8212;all</strong></p>
<blockquote>
<div>Include all nodes in the output files instead of only those active
in the source space. Note that the output files will be huge if
this option is active.</div></blockquote>
</div>
<div class="section" id="listing-bem-mesh-data">
<span id="behbbehj"></span><h2>Listing BEM mesh data<a class="headerlink" href="#listing-bem-mesh-data" title="Permalink to this headline">¶</a></h2>
<p>The utility mne_list_bem outputs
the BEM meshes in text format. The default output data contains
the <em>x</em>, <em>y</em>, and <em>z</em> coordinates
of the vertices, listed in millimeters, one vertex per line.</p>
<p>The command-line options are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;bem &lt;*name*&gt;</strong></p>
<blockquote>
<div>The BEM file to be listed. The file name normally ends with -bem.fif or -bem-sol.fif .</div></blockquote>
<p><strong>&#8212;out &lt;*name*&gt;</strong></p>
<blockquote>
<div>The output file name.</div></blockquote>
<p><strong>&#8212;id &lt;*number*&gt;</strong></p>
<blockquote>
<div>Identify the surface to be listed. The surfaces are numbered starting with
the innermost surface. Thus, for a three-layer model the surface numbers
are: 4 = scalp, 3 = outer skull, 1 = inner skull
Default value is 4.</div></blockquote>
<p><strong>&#8212;gdipoli</strong></p>
<blockquote>
<div>List the surfaces in the format required by Thom Oostendorp&#8217;s
gdipoli program. This is also the default input format for mne_surf2bem .</div></blockquote>
<p><strong>&#8212;meters</strong></p>
<blockquote>
<div>List the surface coordinates in meters instead of millimeters.</div></blockquote>
<p><strong>&#8212;surf</strong></p>
<blockquote>
<div>Write the output in the binary FreeSurfer format.</div></blockquote>
<p><strong>&#8212;xfit</strong></p>
<blockquote>
<div>Write a file compatible with xfit. This is the same effect as using
the options <tt class="docutils literal"><span class="pre">--gdipoli</span></tt> and <tt class="docutils literal"><span class="pre">--meters</span></tt> together.</div></blockquote>
</div>
<div class="section" id="converting-surface-data-between-different-formats">
<span id="behdiajg"></span><h2>Converting surface data between different formats<a class="headerlink" href="#converting-surface-data-between-different-formats" title="Permalink to this headline">¶</a></h2>
<p>The utility mne_convert_surface converts
surface data files between different formats.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The MNE Matlab toolbox functions enable    reading of FreeSurfer surface files directly. Therefore, the <tt class="docutils literal"><span class="pre">--mat</span></tt>   option has been removed. The dfs file format conversion functionality    has been moved here from mne_convert_dfs .    Consequently, mne_convert_dfs has    been removed from MNE software.</p>
</div>
<div class="section" id="command-line-options">
<span id="babeabaa"></span><h3>command-line options<a class="headerlink" href="#command-line-options" title="Permalink to this headline">¶</a></h3>
<p>mne_convert_surface accepts
the following command-line options:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;fif &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a fif format input file. The first surface (source space)
from this file will be read.</div></blockquote>
<p><strong>&#8212;tri &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a text format input file. The format of this file is described in <a class="reference internal" href="forward.html#behdefcd"><em>Tessellation file format</em></a>.</div></blockquote>
<p><strong>&#8212;meters</strong></p>
<blockquote>
<div>The unit of measure for the vertex locations in a text input files
is meters instead of the default millimeters. This option does not
have any effect on the interpretation of the FreeSurfer surface
files specified with the <tt class="docutils literal"><span class="pre">--surf</span></tt> option.</div></blockquote>
<p><strong>&#8212;swap</strong></p>
<blockquote>
<div>Swap the ordering or the triangle vertices. The standard convention in
the MNE software is to have the vertices in text format files ordered
so that the vector cross product of the vectors from vertex 1 to
2 and 1 to 3 gives the direction of the outward surface normal. This
is also called the counterclockwise ordering. If your text input file
does not comply with this right-hand rule, use the <tt class="docutils literal"><span class="pre">--swap</span></tt> option.
This option does not have any effect on the interpretation of the FreeSurfer surface
files specified with the <tt class="docutils literal"><span class="pre">--surf</span></tt> option.</div></blockquote>
<p><strong>&#8212;surf &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a FreeSurfer format
input file.</div></blockquote>
<p><strong>&#8212;dfs &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of a dfs file to be converted. The surfaces produced
by BrainSuite are in the dfs format.</div></blockquote>
<p><strong>&#8212;mghmri &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a mgh/mgz format MRI data file which will be used to define
the coordinate transformation to be applied to the data read from
a dfs file to bring it to the FreeSurfer MRI
coordinates, <em>i.e.</em>, the coordinate system of
the MRI stack in the file. In addition, this option can be used
to insert &#8220;volume geometry&#8221; information to the FreeSurfer
surface file output (<tt class="docutils literal"><span class="pre">--surfout</span></tt> option). If the input file already
contains the volume geometry information, &#8211;replacegeom is needed
to override the input volume geometry and to proceed to writing
the data.</div></blockquote>
<p><strong>&#8212;replacegeom</strong></p>
<blockquote>
<div>Replaces existing volume geometry information. Used in conjunction
with the <tt class="docutils literal"><span class="pre">--mghmri</span></tt> option described above.</div></blockquote>
<p><strong>&#8212;fifmri &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a fif format MRI destription file which will be used to define
the coordinate transformation to be applied to the data read from
a dfs file to bring it to the same coordinate system as the MRI stack
in the file.</div></blockquote>
<p><strong>&#8212;trans &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of a text file which contains the coordinate
transformation to be applied to the data read from the dfs file
to bring it to the MRI coordinates, see below. This option is rarely
needed.</div></blockquote>
<p><strong>&#8212;flip</strong></p>
<blockquote>
<div>By default, the dfs surface nodes are assumed to be in a right-anterior-superior
(RAS) coordinate system with its origin at the left-posterior-inferior
(LPI) corner of the MRI stack. Sometimes the dfs file has left and
right flipped. This option reverses this flip, <em>i.e.</em>,
assumes the surface coordinate system is left-anterior-superior
(LAS) with its origin in the right-posterior-inferior (RPI) corner
of the MRI stack.</div></blockquote>
<p><strong>&#8212;shift &lt;*value/mm*&gt;</strong></p>
<blockquote>
<div>Shift the surface vertices to the direction of the surface normals
by this amount before saving the surface.</div></blockquote>
<p><strong>&#8212;surfout &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a FreeSurfer format output file.</div></blockquote>
<p><strong>&#8212;fifout &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a fif format output file.</div></blockquote>
<p><strong>&#8212;triout &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies an ASCII output file that will contain the surface data
in the triangle file format desribed in <a class="reference internal" href="forward.html#behdefcd"><em>Tessellation file format</em></a>.</div></blockquote>
<p><strong>&#8212;pntout &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a ASCII output file which will contain the vertex numbers only.</div></blockquote>
<p><strong>&#8212;metersout</strong></p>
<blockquote>
<div>With this option the ASCII output will list the vertex coordinates
in meters instead of millimeters.</div></blockquote>
<p><strong>&#8212;swapout</strong></p>
<blockquote>
<div>Defines the vertex ordering of ASCII triangle files to be output.
For details, see <tt class="docutils literal"><span class="pre">--swap</span></tt> option, above.</div></blockquote>
<p><strong>&#8212;smfout &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a smf (Simple Model Format) output file. For details of this
format, see <a class="reference external" href="http://people.sc.fsu.edu/~jburkardt/data/smf/smf.txt">http://people.sc.fsu.edu/~jburkardt/data/smf/smf.txt</a>.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Multiple output options can be specified to    produce outputs in several different formats with a single invocation    of mne_convert_surface .</p>
</div>
<p>The coordinate transformation file specified with the <tt class="docutils literal"><span class="pre">--trans</span></tt> should contain
a 4 x 4 coordinate transformation matrix:</p>
<div class="math">
<p><img src="../_images/math/75c28577be86ba27064981a976e3bc25078a2af1.png" alt="T = \begin{bmatrix}
R_{11} &amp; R_{12} &amp; R_{13} &amp; x_0 \\
R_{13} &amp; R_{13} &amp; R_{13} &amp; y_0 \\
R_{13} &amp; R_{13} &amp; R_{13} &amp; z_0 \\
0 &amp; 0 &amp; 0 &amp; 1
\end{bmatrix}"/></p>
</div><p>defined so that if the augmented location vectors in the
dfs file and MRI coordinate systems are denoted by <img class="math" src="../_images/math/46cef0b83dd2a922acb0a0e854c719c6c48d57bd.png" alt="r_{dfs} = [x_{dfs} y_{dfs} z_{dfs} 1]^T"/> and <img class="math" src="../_images/math/c4c2a974ad3586ad6ed72e026229c69567ede5f5.png" alt="r_{MRI} = [x_{MRI} y_{MRI} z_{MRI} 1]^T"/>,
respectively,</p>
<div class="math">
<p><img src="../_images/math/b667ff3758bea70c415e9fa68c1ec9eea36acfde.png" alt="r_{MRI} = Tr_{dfs}"/></p>
</div></div>
</div>
<div class="section" id="converting-mri-data-into-the-fif-format">
<span id="babbhhhe"></span><h2>Converting MRI data into the fif format<a class="headerlink" href="#converting-mri-data-into-the-fif-format" title="Permalink to this headline">¶</a></h2>
<p>The utility mne_make_cor_set creates
a fif format MRI description
file optionally including the MRI data using FreeSurfer MRI volume
data as input. The command-line options are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;dir &lt;*directory*&gt;</strong></p>
<blockquote>
<div>Specifies a directory containing the MRI volume in COR format. Any
previous <tt class="docutils literal"><span class="pre">--mgh</span></tt> options are cancelled when this option
is encountered.</div></blockquote>
<p><strong>&#8212;withdata</strong></p>
<blockquote>
<div>Include the pixel data to the output file. This option is implied
with the <tt class="docutils literal"><span class="pre">--mgh</span></tt> option.</div></blockquote>
<p><strong>&#8212;mgh &lt;*name*&gt;</strong></p>
<blockquote>
<div>An MRI volume volume file in mgh or mgz format.
The <tt class="docutils literal"><span class="pre">--withdata</span></tt> option is implied with this type of
input. Furthermore, the <img class="math" src="../_images/math/b9994c8683876e2b06655e388d82f02147da76fa.png" alt="T_3"/> transformation,
the Talairach transformation <img class="math" src="../_images/math/342fe728eb5cbbac553873b4bc3b1b70f796d6aa.png" alt="T_4"/> from
the talairach.xfm file referred to in the MRI volume, and the the
fixed transforms <img class="math" src="../_images/math/83b818dacd8416b4f7583e8da655cfb37efbe3d5.png" alt="T_-"/> and <img class="math" src="../_images/math/75924c7f6094d79374678bb1164446951fd80f62.png" alt="T_+"/> will
added to the output file. For definition of the coordinate transformations,
see <a class="reference internal" href="forward.html#chdedfib"><em>MEG/EEG and MRI coordinate systems</em></a>.</div></blockquote>
<p><strong>&#8212;talairach &lt;*name*&gt;</strong></p>
<blockquote>
<div>Take the Talairach transform from this file instead of the one specified
in mgh/mgz files.</div></blockquote>
<p><strong>&#8212;out &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the output file, which is a fif-format MRI description
file.</div></blockquote>
</div>
<div class="section" id="collecting-coordinate-transformations-into-one-file">
<span id="babbifij"></span><h2>Collecting coordinate transformations into one file<a class="headerlink" href="#collecting-coordinate-transformations-into-one-file" title="Permalink to this headline">¶</a></h2>
<p>The utility mne_collect_transforms collects
coordinate transform information from various sources and saves
them into a single fif file. The coordinate transformations used
by MNE software are summarized in Figure 5.1. The output
of mne_collect_transforms may
include all transforms referred to therein except for the sensor
coordinate system transformations <img class="math" src="../_images/math/567b686ff41e3545838b59d8092bc61f7aed2f11.png" alt="T_{s_1} \dotso T_{s_n}"/>.
The command-line options are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;meas &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a measurement data file which provides <img class="math" src="../_images/math/e4934a6da5a5a799606542faefa8cb95336bee82.png" alt="T_1"/>.
A forward solution or an inverse operator file can also be specified
as implied by Table 5.1.</div></blockquote>
<p><strong>&#8212;mri &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies an MRI description or a standalone coordinate transformation
file produced by mne_analyze which
provides <img class="math" src="../_images/math/88e85b7f4f17a081d270a9f62304e6080881e439.png" alt="T_2"/>. If the <tt class="docutils literal"><span class="pre">--mgh</span></tt> option
is not present mne_collect_transforms also
tries to find <img class="math" src="../_images/math/b9994c8683876e2b06655e388d82f02147da76fa.png" alt="T_3"/>, <img class="math" src="../_images/math/342fe728eb5cbbac553873b4bc3b1b70f796d6aa.png" alt="T_4"/>, <img class="math" src="../_images/math/83b818dacd8416b4f7583e8da655cfb37efbe3d5.png" alt="T_-"/>,
and <img class="math" src="../_images/math/75924c7f6094d79374678bb1164446951fd80f62.png" alt="T_+"/> from this file.</div></blockquote>
<p><strong>&#8212;mgh &lt;*name*&gt;</strong></p>
<blockquote>
<div>An MRI volume volume file in mgh or mgz format.
This file provides <img class="math" src="../_images/math/b9994c8683876e2b06655e388d82f02147da76fa.png" alt="T_3"/>. The transformation <img class="math" src="../_images/math/342fe728eb5cbbac553873b4bc3b1b70f796d6aa.png" alt="T_4"/> will
be read from the talairach.xfm file referred to in the MRI volume.
The fixed transforms <img class="math" src="../_images/math/83b818dacd8416b4f7583e8da655cfb37efbe3d5.png" alt="T_-"/> and <img class="math" src="../_images/math/75924c7f6094d79374678bb1164446951fd80f62.png" alt="T_+"/> will
also be created.</div></blockquote>
<p><strong>&#8212;out &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the output file. If this option is not present, the collected transformations
will be output on screen but not saved.</div></blockquote>
</div>
<div class="section" id="converting-an-ncov-covariance-matrix-file-to-fiff">
<span id="behchghd"></span><h2>Converting an ncov covariance matrix file to fiff<a class="headerlink" href="#converting-an-ncov-covariance-matrix-file-to-fiff" title="Permalink to this headline">¶</a></h2>
<p>The ncov file format was used to store the noise-covariance
matrix file. The MNE software requires that the covariance matrix
files are in fif format. The utility mne_convert_ncov converts
ncov files to fif format.</p>
<p>The command-line options are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;ncov &lt;*name*&gt;</strong></p>
<blockquote>
<div>The ncov file to be converted.</div></blockquote>
<p><strong>&#8212;meas &lt;*name*&gt;</strong></p>
<blockquote>
<div>A fif format measurement file used to assign channel names to the noise-covariance
matrix elements. This file should have precisely the same channel
order within MEG and EEG as the ncov file. Typically, both the ncov
file and the measurement file are created by the now mature off-line
averager, meg_average .</div></blockquote>
</div>
<div class="section" id="converting-a-lisp-covariance-matrix-to-fiff">
<span id="behcdbhg"></span><h2>Converting a lisp covariance matrix to fiff<a class="headerlink" href="#converting-a-lisp-covariance-matrix-to-fiff" title="Permalink to this headline">¶</a></h2>
<p>The utility mne_convert_lspcov converts a LISP-format noise-covariance file,
produced by the Neuromag signal processor, graph into fif format.</p>
<p>The command-line options are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;lspcov &lt;*name*&gt;</strong></p>
<blockquote>
<div>The LISP noise-covariance matrix file to be converted.</div></blockquote>
<p><strong>&#8212;meas &lt;*name*&gt;</strong></p>
<blockquote>
<div>A fif format measurement file used to assign channel names to the noise-covariance
matrix elements. This file should have precisely the same channel
order within MEG and EEG as the LISP-format covariance matrix file.</div></blockquote>
<p><strong>&#8212;out &lt;*name*&gt;</strong></p>
<blockquote>
<div>The name of a fif format output file. The file name should end with
-cov.fif.text format output file. No information about the channel names
is included. The covariance matrix file is listed row by row. This
file can be loaded to MATLAB, for example</div></blockquote>
<p><strong>&#8212;outasc &lt;*name*&gt;</strong></p>
<blockquote>
<div>The name of a text format output file. No information about the channel
names is included. The covariance matrix file is listed row by row.
This file can be loaded to MATLAB, for example</div></blockquote>
</div>
<div class="section" id="the-mne-data-file-conversion-tool">
<span id="behccebj"></span><h2>The MNE data file conversion tool<a class="headerlink" href="#the-mne-data-file-conversion-tool" title="Permalink to this headline">¶</a></h2>
<p>This utility, called mne_convert_mne_data ,
allows the conversion of various fif files related to the MNE computations
to other formats. The two principal purposes of this utility are
to facilitate development of new analysis approaches with Matlab
and conversion of the forward model and noise covariance matrix
data into evoked-response type fif files, which can be accessed
and displayed with the Neuromag source modelling software.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Most of the functions of mne_convert_mne_data are    now covered by the MNE Matlab toolbox covered in <a class="reference internal" href="matlab.html#ch-matlab"><em>The Matlab toolbox</em></a>.    This toolbox is recommended to avoid creating additional files occupying    disk space.</p>
</div>
<div class="section" id="behciccf">
<span id="id2"></span><h3>Command-line options<a class="headerlink" href="#behciccf" title="Permalink to this headline">¶</a></h3>
<p>The command-line options recognize
by mne_convert_mne_data are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;fwd &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specity the name of the forward solution file to be converted. Channels
specified with the <tt class="docutils literal"><span class="pre">--bad</span></tt> option will be excluded from
the file.</div></blockquote>
<p><strong>&#8212;fixed</strong></p>
<blockquote>
<div>Convert the forward solution to the fixed-orientation mode before outputting
the converted file. With this option only the field patterns corresponding
to a dipole aligned with the estimated cortex surface normal are
output.</div></blockquote>
<p><strong>&#8212;surfsrc</strong></p>
<blockquote>
<div>When outputting a free-orientation forward model (three orthogonal dipole
components present) rotate the dipole coordinate system at each
source node so that the two tangential dipole components are output
first, followed by the field corresponding to the dipole aligned
with the estimated cortex surface normal. The orientation of the
first two dipole components in the tangential plane is arbitrarily selected
to create an orthogonal coordinate system.</div></blockquote>
<p><strong>&#8212;noiseonly</strong></p>
<blockquote>
<div>When creating a &#8216;measurement&#8217; fif file, do not
output a forward model file, just the noise-covariance matrix.</div></blockquote>
<p><strong>&#8212;senscov &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the fif file containing a sensor covariance matrix to
be included with the output. If no other input files are specified
only the covariance matrix is output</div></blockquote>
<p><strong>&#8212;srccov &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the fif file containing the source covariance matrix to
be included with the output. Only diagonal source covariance files
can be handled at the moment.</div></blockquote>
<p><strong>&#8212;bad &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the file containing the names of the channels to
be omitted, one channel name per line. This does not affect the output
of the inverse operator since the channels have been already selected
when the file was created.</div></blockquote>
<p><strong>&#8212;fif</strong></p>
<blockquote>
<div>Output the forward model and the noise-covariance matrix into &#8216;measurement&#8217; fif
files. The forward model files are tagged with &lt;<em>modalities</em>&gt; <tt class="docutils literal"><span class="pre">-meas-fwd.fif</span></tt> and
the noise-covariance matrix files with &lt;<em>modalities</em>&gt; <tt class="docutils literal"><span class="pre">-meas-cov.fif</span></tt> .
Here, modalities is <tt class="docutils literal"><span class="pre">-meg</span></tt> if MEG is included, <tt class="docutils literal"><span class="pre">-eeg</span></tt> if
EEG is included, and <tt class="docutils literal"><span class="pre">-meg-eeg</span></tt> if both types of signals
are present. The inclusion of modalities is controlled by the <tt class="docutils literal"><span class="pre">--meg</span></tt> and <tt class="docutils literal"><span class="pre">--eeg</span></tt> options.</div></blockquote>
<p><strong>&#8212;mat</strong></p>
<blockquote>
<div>Output the data into MATLAB mat files. This is the default. The
forward model files are tagged with &lt;<em>modalities</em>&gt; <tt class="docutils literal"><span class="pre">-fwd.mat</span></tt> forward model
and noise-covariance matrix output, with <tt class="docutils literal"><span class="pre">-inv.mat</span></tt> for inverse
operator output, and with <tt class="docutils literal"><span class="pre">-inv-meas.mat</span></tt> for combined inverse
operator and measurement data output, respectively. The meaning
of &lt;<em>modalities</em>&gt; is the same
as in the fif output, described above.</div></blockquote>
<p><strong>&#8212;tag &lt;*name*&gt;</strong></p>
<blockquote>
<div>By default, all variables in the matlab output files start with
<tt class="docutils literal"><span class="pre">mne\_</span></tt>. This option allows to change this prefix to &lt;<em>name</em>&gt; _.</div></blockquote>
<p><strong>&#8212;meg</strong></p>
<blockquote>
<div>Include MEG channels from the forward solution and noise-covariance
matrix.</div></blockquote>
<p><strong>&#8212;eeg</strong></p>
<blockquote>
<div>Include EEG channels from the forward solution and noise-covariance
matrix.</div></blockquote>
<p><strong>&#8212;inv &lt;*name*&gt;</strong></p>
<blockquote>
<div>Output the inverse operator data from the specified file into a
mat file. The source and noise covariance matrices as well as active channels
have been previously selected when the inverse operator was created
with mne_inverse_operator . Thus
the options <tt class="docutils literal"><span class="pre">--meg</span></tt> , <tt class="docutils literal"><span class="pre">--eeg</span></tt> , <tt class="docutils literal"><span class="pre">--senscov</span></tt> , <tt class="docutils literal"><span class="pre">--srccov</span></tt> , <tt class="docutils literal"><span class="pre">--noiseonly</span></tt> ,
and <tt class="docutils literal"><span class="pre">--bad</span></tt> do not affect the output of the inverse operator.</div></blockquote>
<p><strong>&#8212;meas &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the file containing measurement data to be output together with
the inverse operator. The channels corresponding to the inverse operator
are automatically selected from the file if <tt class="docutils literal"><span class="pre">--inv</span></tt> .
option is present. Otherwise, the channel selection given with <tt class="docutils literal"><span class="pre">--sel</span></tt> option will
be taken into account.</div></blockquote>
<p><strong>&#8212;set &lt;*number*&gt;</strong></p>
<blockquote>
<div>Select the data set to be output from the measurement file.</div></blockquote>
<p><strong>&#8212;bmin &lt;*value/ms*&gt;</strong></p>
<blockquote>
<div>Specifies the baseline minimum value setting for the measurement signal
output.</div></blockquote>
<p><strong>&#8212;bmax &lt;*value/ms*&gt;</strong></p>
<blockquote>
<div>Specifies the baseline maximum value setting for the measurement signal
output.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The <tt class="docutils literal"><span class="pre">--tmin</span></tt> and <tt class="docutils literal"><span class="pre">--tmax</span></tt> options    which existed in previous versions of mne_converted_mne_data have    been removed. If output of measurement data is requested, the entire    averaged epoch is now included.</p>
</div>
</div>
<div class="section" id="guide-to-combining-options">
<h3>Guide to combining options<a class="headerlink" href="#guide-to-combining-options" title="Permalink to this headline">¶</a></h3>
<p>The combination of options is quite complicated. The <a class="reference internal" href="#behdciii"><em>Guide to combining mne_convert_mne_data options.</em></a> should be
helpful to determine the combination of options appropriate for your needs.</p>
<table border="1" class="docutils" id="behdciii">
<caption>Guide to combining mne_convert_mne_data options.</caption>
<colgroup>
<col width="39%" />
<col width="9%" />
<col width="27%" />
<col width="24%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Desired output</td>
<td>Format</td>
<td>Required options</td>
<td>Optional options</td>
</tr>
<tr class="row-even"><td>forward model</td>
<td>fif</td>
<td>&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;meg and/or &#8212;eeg
&#8212;fif</td>
<td>&#8212;bad &lt;<em>name</em>&gt;
&#8212;surfsrc</td>
</tr>
<tr class="row-odd"><td>forward model</td>
<td>mat</td>
<td>&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;meg and/or &#8211;eeg</td>
<td>&#8212;bad &lt;<em>name</em>&gt;
&#8212;surfsrc</td>
</tr>
<tr class="row-even"><td>forward model and sensor covariance</td>
<td>mat</td>
<td>&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;senscov &lt;<em>name</em>&gt;
&#8212;meg and/or &#8211;eeg</td>
<td>&#8212;bad &lt;<em>name</em>&gt;
&#8212;surfsrc</td>
</tr>
<tr class="row-odd"><td>sensor covariance</td>
<td>fif</td>
<td>&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;senscov &lt;<em>name</em>&gt;
&#8212;noiseonly
&#8212;fif
&#8212;meg and/or &#8211;eeg</td>
<td>&#8212;bad &lt;<em>name</em>&gt;</td>
</tr>
<tr class="row-even"><td>sensor covariance</td>
<td>mat</td>
<td>&#8212;senscov &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;</td>
<td>&#8212;bad &lt;<em>name</em>&gt;</td>
</tr>
<tr class="row-odd"><td>sensor covariance eigenvalues</td>
<td>text</td>
<td>&#8212;senscov &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;
&#8212;eig</td>
<td>&#8212;bad &lt;<em>name</em>&gt;</td>
</tr>
<tr class="row-even"><td>evoked MEG/EEG data</td>
<td>mat</td>
<td>&#8212;meas &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;</td>
<td>&#8212;sel &lt;<em>name</em>&gt;
&#8212;set &lt;<em>number</em>&gt;</td>
</tr>
<tr class="row-odd"><td>evoked MEG/EEG data forward model</td>
<td>mat</td>
<td>&#8212;meas &lt;<em>name</em>&gt;
&#8212;fwd &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;</td>
<td>&#8212;bad &lt;<em>name</em>&gt;
&#8212;set &lt;<em>number</em>&gt;</td>
</tr>
<tr class="row-even"><td>inverse operator data</td>
<td>mat</td>
<td>&#8212;inv &lt;<em>name</em>&gt;
&#8212;out &lt;<em>name</em>&gt;</td>
<td>&nbsp;</td>
</tr>
<tr class="row-odd"><td>inverse operator data evoked
MEG/EEG data</td>
<td>mat</td>
<td>–&#8211;inv &lt;<em>name</em>&gt;
–&#8211;meas &lt;<em>name</em>&gt;
–&#8211;out &lt;<em>name</em>&gt;</td>
<td>&nbsp;</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="matlab-data-structures">
<h3>Matlab data structures<a class="headerlink" href="#matlab-data-structures" title="Permalink to this headline">¶</a></h3>
<p>The Matlab output provided by mne_convert_mne_data is
organized in structures, listed in <a class="reference internal" href="#behcicca"><em>Matlab structures produced by mne_convert_mne_data.</em></a>. The fields
occurring in these structures are listed in <a class="reference internal" href="#babcbigf"><em>The fields of Matlab structures.</em></a>.</p>
<p>The symbols employed in variable size descriptions are:</p>
<p><strong>nloc</strong></p>
<blockquote>
<div>Number
of source locations</div></blockquote>
<p><strong>nsource</strong></p>
<blockquote>
<div>Number
of sources. For fixed orientation sources nsource = nloc whereas nsource = 3*nloc for
free orientation sources</div></blockquote>
<p><strong>nchan</strong></p>
<blockquote>
<div>Number
of measurement channels.</div></blockquote>
<p><strong>ntime</strong></p>
<blockquote>
<div>Number
of time points in the measurement data.</div></blockquote>
<table border="1" class="docutils" id="behcicca">
<caption>Matlab structures produced by mne_convert_mne_data.</caption>
<colgroup>
<col width="28%" />
<col width="72%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Structure</th>
<th class="head">Contents</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>&lt;<em>tag</em>&gt; _meas</td>
<td>Measured data</td>
</tr>
<tr class="row-odd"><td>&lt;<em>tag</em>&gt; _inv</td>
<td>The inverse operator decomposition</td>
</tr>
<tr class="row-even"><td>&lt;<em>tag</em>&gt; _fwd</td>
<td>The forward solution</td>
</tr>
<tr class="row-odd"><td>&lt;<em>tag</em>&gt; _noise</td>
<td>A standalone noise-covariance matrix</td>
</tr>
</tbody>
</table>
<p>The prefix given with the <tt class="docutils literal"><span class="pre">--tag</span></tt> option is indicated &lt;<em>tag</em>&gt; , see <a class="reference internal" href="#behciccf"><em>Command-line options</em></a>. Its default value is MNE.</p>
<table border="1" class="docutils" id="babcbigf">
<caption>The fields of Matlab structures.</caption>
<colgroup>
<col width="23%" />
<col width="17%" />
<col width="60%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Variable</td>
<td>Size</td>
<td>Description</td>
</tr>
<tr class="row-even"><td>fwd</td>
<td>nsource x nchan</td>
<td>The forward solution, one source on each row. For free
orientation sources, the fields of the three orthogonal
dipoles for each location are listed consecutively.</td>
</tr>
<tr class="row-odd"><td>names ch_names</td>
<td>nchan (string)</td>
<td>String array containing the names of the channels included</td>
</tr>
<tr class="row-even"><td>ch_types</td>
<td>nchan x 2</td>
<td>The column lists the types of the channels (1 = MEG,
2 = EEG). The second column lists the coil types, see
<a class="reference internal" href="forward.html#bgbbhgec"><em>Normal coil descriptions. Note: If a plus-minus sign occurs in several coordinates, all possible combinations have to be included.</em></a> and <a class="reference internal" href="forward.html#chdbdfje"><em>Accurate coil descriptions</em></a>. For EEG electrodes,
this value equals one.</td>
</tr>
<tr class="row-odd"><td>ch_pos</td>
<td>nchan x 3</td>
<td>The location information for each channel. The first three
values specify the origin of the sensor coordinate system
or the location of the electrode. For MEG channels, the
following nine number specify the <em>x</em>, <em>y</em>, and
<em>z</em>-direction unit vectors of the sensor coordinate system.
For EEG electrodes the first unit vector specifies the
location of the reference electrode. If the reference is
not specified this value is all zeroes. The remaining unit
vectors are irrelevant for EEG electrodes.</td>
</tr>
<tr class="row-even"><td>ch_lognos</td>
<td>nchan x 1</td>
<td>Logical channel numbers as listed in the fiff file</td>
</tr>
<tr class="row-odd"><td>ch_units</td>
<td>nchan x 2</td>
<td>Units and unit multipliers as listed in the fif file. The
unit of the data is listed in the first column (T = 112,
T/m = 201, V = 107). At present, the second column will be
always zero, <em>i.e.</em>, no unit multiplier.</td>
</tr>
<tr class="row-even"><td>ch_cals</td>
<td>nchan x 2</td>
<td>Even if the data comes from the conversion already
calibrated, the original calibration factors are included.
The first column is the range member of the fif data
structures and while the second is the cal member. To get
calibrated values in the units given in ch_units from the
raw data, the data must be multiplied with the product of
range and cal.</td>
</tr>
<tr class="row-odd"><td>sfreq</td>
<td>1</td>
<td>The sampling frequency in Hz.</td>
</tr>
<tr class="row-even"><td>lowpass</td>
<td>1</td>
<td>Lowpass filter frequency (Hz)</td>
</tr>
<tr class="row-odd"><td>highpass</td>
<td>1</td>
<td>Highpass filter frequency (Hz)</td>
</tr>
<tr class="row-even"><td>source_loc</td>
<td>nloc x 3</td>
<td>The source locations given in the coordinate frame
indicated by the coord_frame member.</td>
</tr>
<tr class="row-odd"><td>source_ori</td>
<td>nsource x 3</td>
<td>The source orientations</td>
</tr>
<tr class="row-even"><td>source_selection</td>
<td>nsource x 2</td>
<td>Indication of the sources selected from the complete source
spaces. Each row contains the number of the source in the
complete source space (starting with 0) and the source
space number (1 or 2). These numbers refer to the order the
two hemispheres where listed when mne_make_source_space was
invoked. mne_setup_source_space lists the left hemisphere
first.</td>
</tr>
<tr class="row-odd"><td>coord_frame</td>
<td>string</td>
<td>Name of the coordinate frame employed in the forward
calculations. Possible values are &#8216;head&#8217; and &#8216;mri&#8217;.</td>
</tr>
<tr class="row-even"><td>mri_head_trans</td>
<td>4 x 4</td>
<td>The coordinate frame transformation from mri the MEG &#8216;head&#8217;
coordinates.</td>
</tr>
<tr class="row-odd"><td>meg_head_trans</td>
<td>4 x 4</td>
<td>The coordinate frame transformation from the MEG device
coordinates to the MEG head coordinates</td>
</tr>
<tr class="row-even"><td>noise_cov</td>
<td>nchan x nchan</td>
<td>The noise covariance matrix</td>
</tr>
<tr class="row-odd"><td>source_cov</td>
<td>nsource</td>
<td>The elements of the diagonal source covariance matrix.</td>
</tr>
<tr class="row-even"><td>sing</td>
<td>nchan</td>
<td>The singular values of
<img class="math" src="../_images/math/76c0dd7964db172c0931617bae02d6a092f464f1.png" alt="A = C_0^{-^1/_2} G R^C = U \Lambda V^T"/>
with <img class="math" src="../_images/math/9d86170e7de539c0ff999de09621ee0c7b6c8ed0.png" alt="R"/> selected so that
<img class="math" src="../_images/math/d4fc520112e766673e255bfe94031f1eb3b056b6.png" alt="\text{trace}(AA^T) / \text{trace}(I) = 1"/>
as discussed in <a class="reference internal" href="mne.html#chddhage"><em>Whitening and scaling</em></a></td>
</tr>
<tr class="row-odd"><td>eigen_fields</td>
<td>nchan x nchan</td>
<td>The rows of this matrix are the left singular vectors of
<img class="math" src="../_images/math/0acafa529182e79b4f56165ec677554fba7fcf98.png" alt="A"/>, i.e., the columns of <img class="math" src="../_images/math/73f5e249c88b2b3068263480f576b051cb5c4f6e.png" alt="U"/>, see above.</td>
</tr>
<tr class="row-even"><td>eigen_leads</td>
<td>nchan x nsource</td>
<td>The rows of this matrix are the right singular vectors of
<img class="math" src="../_images/math/0acafa529182e79b4f56165ec677554fba7fcf98.png" alt="A"/>, i.e., the columns of <img class="math" src="../_images/math/c99df7a209495334da442b1ec998abaabfa320d8.png" alt="V"/>, see above.</td>
</tr>
<tr class="row-odd"><td>noise_eigenval</td>
<td>nchan</td>
<td>In terms of <a class="reference internal" href="mne.html#chddhage"><em>Whitening and scaling</em></a>, eigenvalues of <img class="math" src="../_images/math/e765a3271984fcd2ca5a9ffd3cbe2256ec243f1e.png" alt="C_0"/>,
i.e., not scaled with number of averages.</td>
</tr>
<tr class="row-even"><td>noise_eigenvec</td>
<td>nchan</td>
<td>Eigenvectors of the noise covariance matrix. In terms of
<a class="reference internal" href="mne.html#chddhage"><em>Whitening and scaling</em></a>, <img class="math" src="../_images/math/083aa2c5465519c08dd0a2bacb1c6754ca1fc9f1.png" alt="U_C^T"/>.</td>
</tr>
<tr class="row-odd"><td>data</td>
<td>nchan x ntime</td>
<td>The measured data. One row contains the data at one time
point.</td>
</tr>
<tr class="row-even"><td>times</td>
<td>ntime</td>
<td>The time points in the above matrix in seconds</td>
</tr>
<tr class="row-odd"><td>nave</td>
<td>1</td>
<td>Number of averages as listed in the data file.</td>
</tr>
<tr class="row-even"><td>meas_times</td>
<td>ntime</td>
<td>The time points in seconds.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="converting-raw-data-to-matlab-format">
<span id="convert-to-matlab"></span><h2>Converting raw data to Matlab format<a class="headerlink" href="#converting-raw-data-to-matlab-format" title="Permalink to this headline">¶</a></h2>
<p>The utility mne_raw2mat converts
all or selected channels from a raw data file to a Matlab mat file.
In addition, this utility can provide information about the raw
data file so that the raw data can be read directly from the original
fif file using Matlab file I/O routines.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The MNE Matlab toolbox described in <a class="reference internal" href="matlab.html#ch-matlab"><em>The Matlab toolbox</em></a> provides    direct access to raw fif files without a need for conversion to    mat file format first. Therefore, it is recommended that you use    the Matlab toolbox rather than  mne_raw2mat which    creates large files occupying disk space unnecessarily.</p>
</div>
<div class="section" id="id3">
<h3>Command-line options<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<p>mne_raw2mat accepts the
following command-line options:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;raw &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the raw data fif file to convert.</div></blockquote>
<p><strong>&#8212;mat &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the destination Matlab file.</div></blockquote>
<p><strong>&#8212;info</strong></p>
<blockquote>
<div>With this option present, only information about the raw data file
is included. The raw data itself is omitted.</div></blockquote>
<p><strong>&#8212;sel &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a text file which contains the names of the channels to include
in the output file, one channel name per line. If the <tt class="docutils literal"><span class="pre">--info</span></tt> option
is specified, <tt class="docutils literal"><span class="pre">--sel</span></tt> does not have any effect.</div></blockquote>
<p><strong>&#8212;tag &lt;*tag*&gt;</strong></p>
<blockquote>
<div>By default, all Matlab variables included in the output file start
with <tt class="docutils literal"><span class="pre">mne\_</span></tt>. This option changes the prefix to &lt;<em>tag</em>&gt; _.</div></blockquote>
</div>
<div class="section" id="id4">
<h3>Matlab data structures<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<p>The Matlab files output by mne_raw2mat can
contain two data structures, &lt;<em>tag</em>&gt;_raw and &lt;<em>tag</em>&gt;_raw_info .
If <tt class="docutils literal"><span class="pre">--info</span></tt> option is specifed, the file contains the
latter structure only.</p>
<p>The &lt;<em>tag</em>&gt;_raw structure
contains only one field, data which
is a matrix containing the raw data. Each row of this matrix constitutes
the data from one channel in the original file. The data type of
this matrix is the same of the original data (2-byte signed integer,
4-byte signed integer, or single-precision float).</p>
<p>The fields of the &lt;<em>tag</em>&gt;_raw_info structure
are listed in <a class="reference internal" href="#behfdcih"><em>The fields of the raw data info structure.</em></a>. Further explanation of the bufs field
is provided in <a class="reference internal" href="#behjeihj"><em>The bufs member of the raw data info structure.</em></a>.</p>
<table border="1" class="docutils" id="behfdcih">
<caption>The fields of the raw data info structure.</caption>
<colgroup>
<col width="23%" />
<col width="17%" />
<col width="60%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Variable</td>
<td>Size</td>
<td>Description</td>
</tr>
<tr class="row-even"><td>orig_file</td>
<td>string</td>
<td>The name of the original fif file specified with the
<tt class="docutils literal"><span class="pre">--raw</span></tt> option.</td>
</tr>
<tr class="row-odd"><td>nchan</td>
<td>1</td>
<td>Number of channels.</td>
</tr>
<tr class="row-even"><td>nsamp</td>
<td>1</td>
<td>Total number of samples</td>
</tr>
<tr class="row-odd"><td>bufs</td>
<td>nbuf x 4</td>
<td>This field is present if <tt class="docutils literal"><span class="pre">--info</span></tt> option was specified on
the command line. For details, see <a class="reference internal" href="#behjeihj"><em>The bufs member of the raw data info structure.</em></a>.</td>
</tr>
<tr class="row-even"><td>sfreq</td>
<td>1</td>
<td>The sampling frequency in Hz.</td>
</tr>
<tr class="row-odd"><td>lowpass</td>
<td>1</td>
<td>Lowpass filter frequency (Hz)</td>
</tr>
<tr class="row-even"><td>highpass</td>
<td>1</td>
<td>Highpass filter frequency (Hz)</td>
</tr>
<tr class="row-odd"><td>ch_names</td>
<td>nchan (string)</td>
<td>String array containing the names of the channels included</td>
</tr>
<tr class="row-even"><td>ch_types</td>
<td>nchan x 2</td>
<td>The column lists the types of the channesl (1 = MEG, 2 =
EEG). The second column lists the coil types, see
<a class="reference internal" href="forward.html#bgbbhgec"><em>Normal coil descriptions. Note: If a plus-minus sign occurs in several coordinates, all possible combinations have to be included.</em></a> and <a class="reference internal" href="forward.html#chdbdfje"><em>Accurate coil descriptions</em></a>. For EEG electrodes,
this value equals one.</td>
</tr>
<tr class="row-odd"><td>ch_lognos</td>
<td>nchan x 1</td>
<td>Logical channel numbers as listed in the fiff file</td>
</tr>
<tr class="row-even"><td>ch_units</td>
<td>nchan x 2</td>
<td>Units and unit multipliers as listed in the fif file.
The unit of the data is listed in the first column
(T = 112, T/m = 201, V = 107). At present, the second
column will be always zero, <em>i.e.</em>, no unit multiplier.</td>
</tr>
<tr class="row-odd"><td>ch_pos</td>
<td>nchan x 12</td>
<td>The location information for each channel. The first three
values specify the origin of the sensor coordinate system
or the location of the electrode. For MEG channels, the
following nine number specify the <em>x</em>, <em>y</em>, and
<em>z</em>-direction unit vectors of the sensor coordinate system.
For EEG electrodes the first vector after the electrode
location specifies the location of the reference electrode.
If the reference is not specified this value is all zeroes.
The remaining unit vectors are irrelevant for EEG
electrodes.</td>
</tr>
<tr class="row-even"><td>ch_cals</td>
<td>nchan x 2</td>
<td>The raw data output by mne_raw2mat is uncalibrated.
The first column is the range member of the fiff data
structures and while the second is the cal member. To get
calibrared data values in the units given in ch_units from
the raw data, the data must be multiplied with the product
of range and cal .</td>
</tr>
<tr class="row-odd"><td>meg_head_trans</td>
<td>4 x 4</td>
<td>The coordinate frame transformation from the MEG device
coordinates to the MEG head coordinates.</td>
</tr>
</tbody>
</table>
<table border="1" class="docutils" id="behjeihj">
<caption>The bufs member of the raw data info structure.</caption>
<colgroup>
<col width="24%" />
<col width="76%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Column</td>
<td>Contents</td>
</tr>
<tr class="row-even"><td>1</td>
<td>The raw data type (2 or 16 = 2-byte signed integer, 3 = 4-byte signed
integer, 4 = single-precision float). All data in the fif file are
written in the big-endian byte order. The raw data are stored sample by
sample.</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>Byte location of this buffer in the original fif file.</td>
</tr>
<tr class="row-even"><td>3</td>
<td>First sample of this buffer. Since raw data storing can be switched on
and off during the acquisition, there might be gaps between the end of
one buffer and the beginning of the next.</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>Number of samples in the buffer.</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="converting-epochs-to-matlab-format">
<span id="behfidcb"></span><h2>Converting epochs to Matlab format<a class="headerlink" href="#converting-epochs-to-matlab-format" title="Permalink to this headline">¶</a></h2>
<p>The utility mne_epochs2mat converts
epoch data including all or selected channels from a raw data file
to a simple binary file with an associated description file in Matlab
mat file format. With help of the description file, a matlab program
can easily read the epoch data from the simple binary file. Signal
space projection and bandpass filtering can be optionally applied
to the raw data prior to saving the epochs.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The MNE Matlab toolbox described in <a class="reference internal" href="matlab.html#ch-matlab"><em>The Matlab toolbox</em></a> provides direct    access to raw fif files without conversion with mne_epochs2mat first.    Therefore, it is recommended that you use the Matlab toolbox rather than mne_epochs2mat which    creates large files occupying disk space unnecessarily. An exception    to this is the case where you apply a filter to the data and save    the band-pass filtered epochs.</p>
</div>
<div class="section" id="id5">
<h3>Command-line options<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<p>mne_epochs2mat accepts
the following command-line options are:</p>
<p><strong>&#8212;version</strong></p>
<blockquote>
<div>Show the program version and compilation date.</div></blockquote>
<p><strong>&#8212;help</strong></p>
<blockquote>
<div>List the command-line options.</div></blockquote>
<p><strong>&#8212;raw &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the raw data fif file to use as input.</div></blockquote>
<p><strong>&#8212;mat &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies the name of the destination file. Anything following the last
period in the file name will be removed before composing the output
file name. The binary epoch file will be called &lt;<em>trimmed name</em>&gt; <tt class="docutils literal"><span class="pre">.epochs</span></tt> and
the corresponding Matlab description file will be &lt;<em>trimmed name</em>&gt; <tt class="docutils literal"><span class="pre">_desc.mat</span></tt> .</div></blockquote>
<p><strong>&#8212;tag &lt;*tag*&gt;</strong></p>
<blockquote>
<div>By default, all Matlab variables included in the description file
start with <tt class="docutils literal"><span class="pre">mne\_</span></tt>. This option changes the prefix to &lt;<em>tag</em>&gt; _.</div></blockquote>
<p><strong>&#8212;events &lt;*name*&gt;</strong></p>
<blockquote>
<div>The file containing the event definitions. This can be a text or
fif format file produced by mne_process_raw or mne_browse_raw ,
see <a class="reference internal" href="browse.html#cacbcegc"><em>Event files</em></a>. With help of this file it is possible
to select virtually any data segment from the raw data file. If
this option is missing, the digital trigger channel in the raw data
file or a fif format event file produced automatically by mne_process_raw or mne_browse_raw is
consulted for event information.</div></blockquote>
<p><strong>&#8212;event &lt;*name*&gt;</strong></p>
<blockquote>
<div>Event number identifying the epochs of interest.</div></blockquote>
<p><strong>&#8212;tmin &lt;*time/ms*&gt;</strong></p>
<blockquote>
<div>The starting point of the epoch with respect to the event of interest.</div></blockquote>
<p><strong>&#8212;tmax &lt;*time/ms*&gt;</strong></p>
<blockquote>
<div>The endpoint of the epoch with respect to the event of interest.</div></blockquote>
<p><strong>&#8212;sel &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies a text file which contains the names of the channels to include
in the output file, one channel name per line. If the <tt class="docutils literal"><span class="pre">--inv</span></tt> option
is specified, <tt class="docutils literal"><span class="pre">--sel</span></tt> is ignored. If neither <tt class="docutils literal"><span class="pre">--inv</span></tt> nor <tt class="docutils literal"><span class="pre">--sel</span></tt> is
present, all MEG and EEG channels are included. The digital trigger
channel can be included with the <tt class="docutils literal"><span class="pre">--includetrig</span></tt> option, described
below.</div></blockquote>
<p><strong>&#8212;inv &lt;*name*&gt;</strong></p>
<blockquote>
<div>Specifies an inverse operator, which will be employed in two ways. First,
the channels included to output will be those included in the inverse
operator. Second, any signal-space projection operator present in
the inverse operator file will be applied to the data. This option
cancels the effect of <tt class="docutils literal"><span class="pre">--sel</span></tt> and <tt class="docutils literal"><span class="pre">--proj</span></tt> options.</div></blockquote>
<p><strong>&#8212;digtrig &lt;*name*&gt;</strong></p>
<blockquote>
<div>Name of the composite digital trigger channel. The default value
is &#8216;STI 014&#8217;. Underscores in the channel name
will be replaced by spaces.</div></blockquote>
<p><strong>&#8212;digtrigmask &lt;*number*&gt;</strong></p>
<blockquote>
<div>Mask to be applied to the trigger channel values before considering them.
This option is useful if one wants to set some bits in a don&#8217;t care
state. For example, some finger response pads keep the trigger lines
high if not in use, <em>i.e.</em>, a finger is not in
place. Yet, it is convenient to keep these devices permanently connected
to the acquisition system. The number can be given in decimal or
hexadecimal format (beginning with 0x or 0X). For example, the value
255 (0xFF) means that only the lowest order byte (usually trigger
lines 1 - 8 or bits 0 - 7) will be considered.</div></blockquote>
<p><strong>&#8212;includetrig</strong></p>
<blockquote>
<div>Add the digital trigger channel to the list of channels to output.
This option should not be used if the trigger channel is already
included in the selection specified with the <tt class="docutils literal"><span class="pre">--sel</span></tt> option.</div></blockquote>
<p><strong>&#8212;filtersize &lt;*size*&gt;</strong></p>
<blockquote>
<div>Adjust the length of the FFT to be applied in filtering. The number will
be rounded up to the next power of two. If the size is <img class="math" src="../_images/math/75e27f04188974063be3230dca208cd495b77ce1.png" alt="N"/>,
the corresponding length of time is <img class="math" src="../_images/math/5d7966cb43437da0da0f08d760fb0b011f628388.png" alt="^N/_{f_s}"/>,
where <img class="math" src="../_images/math/a2750eb24a00e1c35b2e560ee2981dbcb52d7b3c.png" alt="f_s"/> is the sampling frequency
of your data. The filtering procedure includes overlapping tapers
of length <img class="math" src="../_images/math/454b68d1c26666336dabead9b7a0820684d523f8.png" alt="^N/_2"/> so that the total FFT
length will actually be <img class="math" src="../_images/math/fa6e4221bb72e2ef97c0ca409174b8d3367cfa36.png" alt="2N"/>. The default
value is 4096.</div></blockquote>
<p><strong>&#8212;highpass &lt;*value/Hz*&gt;</strong></p>
<blockquote>
<div>Highpass filter frequency limit. If this is too low with respect
to the selected FFT length and data file sampling frequency, the
data will not be highpass filtered. You can experiment with the
interactive version to find the lowest applicable filter for your
data. This value can be adjusted in the interactive version of the
program. The default is 0, i.e., no highpass filter in effect.</div></blockquote>
<p><strong>&#8212;highpassw &lt;*value/Hz*&gt;</strong></p>
<blockquote>
<div>The width of the transition band of the highpass filter. The default
is 6 frequency bins, where one bin is <img class="math" src="../_images/math/268ab00a71334945c0d42e54331f79fcbe05d559.png" alt="^{f_s}/_{(2N)}"/>.</div></blockquote>
<p><strong>&#8212;lowpass &lt;*value/Hz*&gt;</strong></p>
<blockquote>
<div>Lowpass filter frequency limit. This value can be adjusted in the interactive
version of the program. The default is 40 Hz.</div></blockquote>
<p><strong>&#8212;lowpassw &lt;*value/Hz*&gt;</strong></p>
<blockquote>
<div>The width of the transition band of the lowpass filter. This value
can be adjusted in the interactive version of the program. The default
is 5 Hz.</div></blockquote>
<p><strong>&#8212;filteroff</strong></p>
<blockquote>
<div>Do not filter the data.</div></blockquote>
<p><strong>&#8212;proj &lt;*name*&gt;</strong></p>
<blockquote>
<div>Include signal-space projection (SSP) information from this file.
If the <tt class="docutils literal"><span class="pre">--inv</span></tt> option is present, <tt class="docutils literal"><span class="pre">--proj</span></tt> has
no effect.</div></blockquote>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Baseline has not been subtracted from the epochs. This has to be done in subsequent processing with Matlab if so desired.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Strictly speaking, trigger mask value zero would mean that all trigger inputs are ignored. However, for convenience,    setting the mask to zero or not setting it at all has the same effect    as 0xFFFFFFFF, <em>i.e.</em>, all bits set.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel can also be set with the MNE_TRIGGER_CH_NAME environment variable. Underscores in the variable    value will <em>not</em> be replaced with spaces by mne_browse_raw or mne_process_raw .    Using the <tt class="docutils literal"><span class="pre">--digtrig</span></tt> option supersedes the MNE_TRIGGER_CH_NAME    environment variable.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The digital trigger channel mask can also be    set with the MNE_TRIGGER_CH_MASK environment variable. Using the <tt class="docutils literal"><span class="pre">--digtrigmask</span></tt> option    supersedes the MNE_TRIGGER_CH_MASK environment variable.</p>
</div>
</div>
<div class="section" id="the-binary-epoch-data-file">
<h3>The binary epoch data file<a class="headerlink" href="#the-binary-epoch-data-file" title="Permalink to this headline">¶</a></h3>
<p>mne_epochs2mat saves the
epoch data extracted from the raw data file is a simple binary file.
The data are stored as big-endian single-precision floating point
numbers. Assuming that each of the total of <img class="math" src="../_images/math/3eca8557203e86160952e1c0f735f7417f3285b1.png" alt="p"/> epochs
contains <img class="math" src="../_images/math/413f8a8e40062a9090d9d50b88bc7b551b314c26.png" alt="n"/> channels and <img class="math" src="../_images/math/c4bb40dd65eae6c11b325989b14e0b8d35e4e3ef.png" alt="m"/> time
points, the data <img class="math" src="../_images/math/895511fbbc1e04da635f185e558090fc4f19318d.png" alt="s_{jkl}"/> are ordered
as</p>
<div class="math">
<p><img src="../_images/math/32fa7fb0f3e780888de878e1269b003133c1f2f0.png" alt="s_{111} \dotso s_{1n1} s_{211} \dotso s_{mn1} \dotso s_{mnp}\ ,"/></p>
</div><p>where the first index stands for the time point, the second
for the channel, and the third for the epoch number, respectively.
The data are not calibrated, i.e., the calibration factors present
in the Matlab description file have to be applied to get to physical
units as described below.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">The maximum size of an epoch data file is 2 Gbytes, <em>i.e.</em>, 0.5 Gsamples.</p>
</div>
</div>
<div class="section" id="id6">
<h3>Matlab data structures<a class="headerlink" href="#id6" title="Permalink to this headline">¶</a></h3>
<p>The Matlab description files output by mne_epochs2mat contain
a data structure &lt;<em>tag</em>&gt;_epoch_info .
The fields of the this structure are listed in <a class="reference internal" href="#behfdcih"><em>The fields of the raw data info structure.</em></a>.
Further explanation of the epochs member
is provided in <a class="reference internal" href="#behhaghe"><em>The epochs member of the raw data info structure.</em></a>.</p>
<table border="1" class="docutils" id="behifjij">
<caption>The fields of the raw data info structure.</caption>
<colgroup>
<col width="23%" />
<col width="17%" />
<col width="60%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Variable</td>
<td>Size</td>
<td>Description</td>
</tr>
<tr class="row-even"><td>orig_file</td>
<td>string</td>
<td>The name of the original fif file specified with the
<tt class="docutils literal"><span class="pre">--raw</span></tt> option.</td>
</tr>
<tr class="row-odd"><td>epoch_file</td>
<td>string</td>
<td>The name of the epoch data file produced by mne_epocs2mat.</td>
</tr>
<tr class="row-even"><td>nchan</td>
<td>1</td>
<td>Number of channels.</td>
</tr>
<tr class="row-odd"><td>nepoch</td>
<td>1</td>
<td>Total number of epochs.</td>
</tr>
<tr class="row-even"><td>epochs</td>
<td>nepoch x 5</td>
<td>Description of the content of the epoch data file,
see <a class="reference internal" href="#behhaghe"><em>The epochs member of the raw data info structure.</em></a>.</td>
</tr>
<tr class="row-odd"><td>sfreq</td>
<td>1</td>
<td>The sampling frequency in Hz.</td>
</tr>
<tr class="row-even"><td>lowpass</td>
<td>1</td>
<td>Lowpass filter frequency (Hz)</td>
</tr>
<tr class="row-odd"><td>highpass</td>
<td>1</td>
<td>Highpass filter frequency (Hz)</td>
</tr>
<tr class="row-even"><td>ch_names</td>
<td>nchan (string)</td>
<td>String array containing the names of the channels included</td>
</tr>
<tr class="row-odd"><td>ch_types</td>
<td>nchan x 2</td>
<td>The column lists the types of the channels (1 = MEG, 2 =
EEG). The second column lists the coil types, see
<a class="reference internal" href="forward.html#bgbbhgec"><em>Normal coil descriptions. Note: If a plus-minus sign occurs in several coordinates, all possible combinations have to be included.</em></a> and <a class="reference internal" href="forward.html#chdbdfje"><em>Accurate coil descriptions</em></a>. For EEG electrodes,
this value equals one.</td>
</tr>
<tr class="row-even"><td>ch_lognos</td>
<td>nchan x 1</td>
<td>Logical channel numbers as listed in the fiff file</td>
</tr>
<tr class="row-odd"><td>ch_units</td>
<td>nchan x 2</td>
<td>Units and unit multipliers as listed in the fif file.
The unit of the data is listed in the first column
(T = 112, T/m = 201, V = 107). At present, the second
column will be always zero, <em>i.e.</em>, no unit multiplier.</td>
</tr>
<tr class="row-even"><td>ch_pos</td>
<td>nchan x 12</td>
<td>The location information for each channel. The first three
values specify the origin of the sensor coordinate system
or the location of the electrode. For MEG channels, the
following nine number specify the <em>x</em>, <em>y</em>, and
<em>z</em>-direction unit vectors of the sensor coordinate
system. For EEG electrodes the first vector after the
electrode location specifies the location of the reference
electrode. If the reference is not specified this value is
all zeroes. The remaining unit vectors are irrelevant for
EEG electrodes.</td>
</tr>
<tr class="row-odd"><td>ch_cals</td>
<td>nchan x 2</td>
<td>The raw data output by mne_raw2mat are not calibrated.
The first column is the range member of the fiff data
structures and while the second is the cal member. To
get calibrated data values in the units given in
ch_units from the raw data, the data must be multiplied
with the product of range and cal .</td>
</tr>
<tr class="row-even"><td>meg_head_trans</td>
<td>4 x 4</td>
<td>The coordinate frame transformation from the MEG device
coordinates to the MEG head coordinates.</td>
</tr>
</tbody>
</table>
<table border="1" class="docutils" id="behhaghe">
<caption>The epochs member of the raw data info structure.</caption>
<colgroup>
<col width="19%" />
<col width="81%" />
</colgroup>
<tbody valign="top">
<tr class="row-odd"><td>Column</td>
<td>Contents</td>
</tr>
<tr class="row-even"><td>1</td>
<td>The raw data type (2 or 16 = 2-byte signed integer, 3 = 4-byte
signed integer, 4 = single-precision float). The epoch data are
written using the big-endian byte order. The data are stored
sample by sample.</td>
</tr>
<tr class="row-odd"><td>2</td>
<td>Byte location of this epoch in the binary epoch file.</td>
</tr>
<tr class="row-even"><td>3</td>
<td>First sample of this epoch in the original raw data file.</td>
</tr>
<tr class="row-odd"><td>4</td>
<td>First sample of the epoch with respect to the event.</td>
</tr>
<tr class="row-even"><td>5</td>
<td>Number of samples in the epoch.</td>
</tr>
</tbody>
</table>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">For source modelling purposes, it is recommended    that the MNE Matlab toolbox, see <a class="reference internal" href="matlab.html#ch-matlab"><em>The Matlab toolbox</em></a> is employed    to read the measurement info instead of using the channel information    in the raw data info structure described in <a class="reference internal" href="#behifjij"><em>The fields of the raw data info structure.</em></a>.</p>
</div>
</div>
</div>
</div>


          </div>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
        &copy; Copyright 2012-2015, MNE Developers.
      Last updated on May 14, 2015.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.2.2.
    </div>
  </body>
</html>